// nitro/dist/_libs/confbox.mjs
import { n as __exportAll } from "../_common.mjs";
import { Q as N$1, X as h$1, Z as C$2 } from "../_build/common.mjs";

//#region node_modules/.pnpm/confbox@0.2.2/node_modules/confbox/dist/json5.mjs
var json5_exports = /* @__PURE__ */ __exportAll({ parseJSON5: () => Cu });
function R$2(C$3) {
	return C$3 && C$3.__esModule && Object.prototype.hasOwnProperty.call(C$3, "default") ? C$3.default : C$3;
}
var O$1 = {}, M$2;
function Y$1() {
	return M$2 || (M$2 = 1, O$1.Space_Separator = /[\u1680\u2000-\u200A\u202F\u205F\u3000]/, O$1.ID_Start = /[\xAA\xB5\xBA\xC0-\xD6\xD8-\xF6\xF8-\u02C1\u02C6-\u02D1\u02E0-\u02E4\u02EC\u02EE\u0370-\u0374\u0376\u0377\u037A-\u037D\u037F\u0386\u0388-\u038A\u038C\u038E-\u03A1\u03A3-\u03F5\u03F7-\u0481\u048A-\u052F\u0531-\u0556\u0559\u0561-\u0587\u05D0-\u05EA\u05F0-\u05F2\u0620-\u064A\u066E\u066F\u0671-\u06D3\u06D5\u06E5\u06E6\u06EE\u06EF\u06FA-\u06FC\u06FF\u0710\u0712-\u072F\u074D-\u07A5\u07B1\u07CA-\u07EA\u07F4\u07F5\u07FA\u0800-\u0815\u081A\u0824\u0828\u0840-\u0858\u0860-\u086A\u08A0-\u08B4\u08B6-\u08BD\u0904-\u0939\u093D\u0950\u0958-\u0961\u0971-\u0980\u0985-\u098C\u098F\u0990\u0993-\u09A8\u09AA-\u09B0\u09B2\u09B6-\u09B9\u09BD\u09CE\u09DC\u09DD\u09DF-\u09E1\u09F0\u09F1\u09FC\u0A05-\u0A0A\u0A0F\u0A10\u0A13-\u0A28\u0A2A-\u0A30\u0A32\u0A33\u0A35\u0A36\u0A38\u0A39\u0A59-\u0A5C\u0A5E\u0A72-\u0A74\u0A85-\u0A8D\u0A8F-\u0A91\u0A93-\u0AA8\u0AAA-\u0AB0\u0AB2\u0AB3\u0AB5-\u0AB9\u0ABD\u0AD0\u0AE0\u0AE1\u0AF9\u0B05-\u0B0C\u0B0F\u0B10\u0B13-\u0B28\u0B2A-\u0B30\u0B32\u0B33\u0B35-\u0B39\u0B3D\u0B5C\u0B5D\u0B5F-\u0B61\u0B71\u0B83\u0B85-\u0B8A\u0B8E-\u0B90\u0B92-\u0B95\u0B99\u0B9A\u0B9C\u0B9E\u0B9F\u0BA3\u0BA4\u0BA8-\u0BAA\u0BAE-\u0BB9\u0BD0\u0C05-\u0C0C\u0C0E-\u0C10\u0C12-\u0C28\u0C2A-\u0C39\u0C3D\u0C58-\u0C5A\u0C60\u0C61\u0C80\u0C85-\u0C8C\u0C8E-\u0C90\u0C92-\u0CA8\u0CAA-\u0CB3\u0CB5-\u0CB9\u0CBD\u0CDE\u0CE0\u0CE1\u0CF1\u0CF2\u0D05-\u0D0C\u0D0E-\u0D10\u0D12-\u0D3A\u0D3D\u0D4E\u0D54-\u0D56\u0D5F-\u0D61\u0D7A-\u0D7F\u0D85-\u0D96\u0D9A-\u0DB1\u0DB3-\u0DBB\u0DBD\u0DC0-\u0DC6\u0E01-\u0E30\u0E32\u0E33\u0E40-\u0E46\u0E81\u0E82\u0E84\u0E87\u0E88\u0E8A\u0E8D\u0E94-\u0E97\u0E99-\u0E9F\u0EA1-\u0EA3\u0EA5\u0EA7\u0EAA\u0EAB\u0EAD-\u0EB0\u0EB2\u0EB3\u0EBD\u0EC0-\u0EC4\u0EC6\u0EDC-\u0EDF\u0F00\u0F40-\u0F47\u0F49-\u0F6C\u0F88-\u0F8C\u1000-\u102A\u103F\u1050-\u1055\u105A-\u105D\u1061\u1065\u1066\u106E-\u1070\u1075-\u1081\u108E\u10A0-\u10C5\u10C7\u10CD\u10D0-\u10FA\u10FC-\u1248\u124A-\u124D\u1250-\u1256\u1258\u125A-\u125D\u1260-\u1288\u128A-\u128D\u1290-\u12B0\u12B2-\u12B5\u12B8-\u12BE\u12C0\u12C2-\u12C5\u12C8-\u12D6\u12D8-\u1310\u1312-\u1315\u1318-\u135A\u1380-\u138F\u13A0-\u13F5\u13F8-\u13FD\u1401-\u166C\u166F-\u167F\u1681-\u169A\u16A0-\u16EA\u16EE-\u16F8\u1700-\u170C\u170E-\u1711\u1720-\u1731\u1740-\u1751\u1760-\u176C\u176E-\u1770\u1780-\u17B3\u17D7\u17DC\u1820-\u1877\u1880-\u1884\u1887-\u18A8\u18AA\u18B0-\u18F5\u1900-\u191E\u1950-\u196D\u1970-\u1974\u1980-\u19AB\u19B0-\u19C9\u1A00-\u1A16\u1A20-\u1A54\u1AA7\u1B05-\u1B33\u1B45-\u1B4B\u1B83-\u1BA0\u1BAE\u1BAF\u1BBA-\u1BE5\u1C00-\u1C23\u1C4D-\u1C4F\u1C5A-\u1C7D\u1C80-\u1C88\u1CE9-\u1CEC\u1CEE-\u1CF1\u1CF5\u1CF6\u1D00-\u1DBF\u1E00-\u1F15\u1F18-\u1F1D\u1F20-\u1F45\u1F48-\u1F4D\u1F50-\u1F57\u1F59\u1F5B\u1F5D\u1F5F-\u1F7D\u1F80-\u1FB4\u1FB6-\u1FBC\u1FBE\u1FC2-\u1FC4\u1FC6-\u1FCC\u1FD0-\u1FD3\u1FD6-\u1FDB\u1FE0-\u1FEC\u1FF2-\u1FF4\u1FF6-\u1FFC\u2071\u207F\u2090-\u209C\u2102\u2107\u210A-\u2113\u2115\u2119-\u211D\u2124\u2126\u2128\u212A-\u212D\u212F-\u2139\u213C-\u213F\u2145-\u2149\u214E\u2160-\u2188\u2C00-\u2C2E\u2C30-\u2C5E\u2C60-\u2CE4\u2CEB-\u2CEE\u2CF2\u2CF3\u2D00-\u2D25\u2D27\u2D2D\u2D30-\u2D67\u2D6F\u2D80-\u2D96\u2DA0-\u2DA6\u2DA8-\u2DAE\u2DB0-\u2DB6\u2DB8-\u2DBE\u2DC0-\u2DC6\u2DC8-\u2DCE\u2DD0-\u2DD6\u2DD8-\u2DDE\u2E2F\u3005-\u3007\u3021-\u3029\u3031-\u3035\u3038-\u303C\u3041-\u3096\u309D-\u309F\u30A1-\u30FA\u30FC-\u30FF\u3105-\u312E\u3131-\u318E\u31A0-\u31BA\u31F0-\u31FF\u3400-\u4DB5\u4E00-\u9FEA\uA000-\uA48C\uA4D0-\uA4FD\uA500-\uA60C\uA610-\uA61F\uA62A\uA62B\uA640-\uA66E\uA67F-\uA69D\uA6A0-\uA6EF\uA717-\uA71F\uA722-\uA788\uA78B-\uA7AE\uA7B0-\uA7B7\uA7F7-\uA801\uA803-\uA805\uA807-\uA80A\uA80C-\uA822\uA840-\uA873\uA882-\uA8B3\uA8F2-\uA8F7\uA8FB\uA8FD\uA90A-\uA925\uA930-\uA946\uA960-\uA97C\uA984-\uA9B2\uA9CF\uA9E0-\uA9E4\uA9E6-\uA9EF\uA9FA-\uA9FE\uAA00-\uAA28\uAA40-\uAA42\uAA44-\uAA4B\uAA60-\uAA76\uAA7A\uAA7E-\uAAAF\uAAB1\uAAB5\uAAB6\uAAB9-\uAABD\uAAC0\uAAC2\uAADB-\uAADD\uAAE0-\uAAEA\uAAF2-\uAAF4\uAB01-\uAB06\uAB09-\uAB0E\uAB11-\uAB16\uAB20-\uAB26\uAB28-\uAB2E\uAB30-\uAB5A\uAB5C-\uAB65\uAB70-\uABE2\uAC00-\uD7A3\uD7B0-\uD7C6\uD7CB-\uD7FB\uF900-\uFA6D\uFA70-\uFAD9\uFB00-\uFB06\uFB13-\uFB17\uFB1D\uFB1F-\uFB28\uFB2A-\uFB36\uFB38-\uFB3C\uFB3E\uFB40\uFB41\uFB43\uFB44\uFB46-\uFBB1\uFBD3-\uFD3D\uFD50-\uFD8F\uFD92-\uFDC7\uFDF0-\uFDFB\uFE70-\uFE74\uFE76-\uFEFC\uFF21-\uFF3A\uFF41-\uFF5A\uFF66-\uFFBE\uFFC2-\uFFC7\uFFCA-\uFFCF\uFFD2-\uFFD7\uFFDA-\uFFDC]|\uD800[\uDC00-\uDC0B\uDC0D-\uDC26\uDC28-\uDC3A\uDC3C\uDC3D\uDC3F-\uDC4D\uDC50-\uDC5D\uDC80-\uDCFA\uDD40-\uDD74\uDE80-\uDE9C\uDEA0-\uDED0\uDF00-\uDF1F\uDF2D-\uDF4A\uDF50-\uDF75\uDF80-\uDF9D\uDFA0-\uDFC3\uDFC8-\uDFCF\uDFD1-\uDFD5]|\uD801[\uDC00-\uDC9D\uDCB0-\uDCD3\uDCD8-\uDCFB\uDD00-\uDD27\uDD30-\uDD63\uDE00-\uDF36\uDF40-\uDF55\uDF60-\uDF67]|\uD802[\uDC00-\uDC05\uDC08\uDC0A-\uDC35\uDC37\uDC38\uDC3C\uDC3F-\uDC55\uDC60-\uDC76\uDC80-\uDC9E\uDCE0-\uDCF2\uDCF4\uDCF5\uDD00-\uDD15\uDD20-\uDD39\uDD80-\uDDB7\uDDBE\uDDBF\uDE00\uDE10-\uDE13\uDE15-\uDE17\uDE19-\uDE33\uDE60-\uDE7C\uDE80-\uDE9C\uDEC0-\uDEC7\uDEC9-\uDEE4\uDF00-\uDF35\uDF40-\uDF55\uDF60-\uDF72\uDF80-\uDF91]|\uD803[\uDC00-\uDC48\uDC80-\uDCB2\uDCC0-\uDCF2]|\uD804[\uDC03-\uDC37\uDC83-\uDCAF\uDCD0-\uDCE8\uDD03-\uDD26\uDD50-\uDD72\uDD76\uDD83-\uDDB2\uDDC1-\uDDC4\uDDDA\uDDDC\uDE00-\uDE11\uDE13-\uDE2B\uDE80-\uDE86\uDE88\uDE8A-\uDE8D\uDE8F-\uDE9D\uDE9F-\uDEA8\uDEB0-\uDEDE\uDF05-\uDF0C\uDF0F\uDF10\uDF13-\uDF28\uDF2A-\uDF30\uDF32\uDF33\uDF35-\uDF39\uDF3D\uDF50\uDF5D-\uDF61]|\uD805[\uDC00-\uDC34\uDC47-\uDC4A\uDC80-\uDCAF\uDCC4\uDCC5\uDCC7\uDD80-\uDDAE\uDDD8-\uDDDB\uDE00-\uDE2F\uDE44\uDE80-\uDEAA\uDF00-\uDF19]|\uD806[\uDCA0-\uDCDF\uDCFF\uDE00\uDE0B-\uDE32\uDE3A\uDE50\uDE5C-\uDE83\uDE86-\uDE89\uDEC0-\uDEF8]|\uD807[\uDC00-\uDC08\uDC0A-\uDC2E\uDC40\uDC72-\uDC8F\uDD00-\uDD06\uDD08\uDD09\uDD0B-\uDD30\uDD46]|\uD808[\uDC00-\uDF99]|\uD809[\uDC00-\uDC6E\uDC80-\uDD43]|[\uD80C\uD81C-\uD820\uD840-\uD868\uD86A-\uD86C\uD86F-\uD872\uD874-\uD879][\uDC00-\uDFFF]|\uD80D[\uDC00-\uDC2E]|\uD811[\uDC00-\uDE46]|\uD81A[\uDC00-\uDE38\uDE40-\uDE5E\uDED0-\uDEED\uDF00-\uDF2F\uDF40-\uDF43\uDF63-\uDF77\uDF7D-\uDF8F]|\uD81B[\uDF00-\uDF44\uDF50\uDF93-\uDF9F\uDFE0\uDFE1]|\uD821[\uDC00-\uDFEC]|\uD822[\uDC00-\uDEF2]|\uD82C[\uDC00-\uDD1E\uDD70-\uDEFB]|\uD82F[\uDC00-\uDC6A\uDC70-\uDC7C\uDC80-\uDC88\uDC90-\uDC99]|\uD835[\uDC00-\uDC54\uDC56-\uDC9C\uDC9E\uDC9F\uDCA2\uDCA5\uDCA6\uDCA9-\uDCAC\uDCAE-\uDCB9\uDCBB\uDCBD-\uDCC3\uDCC5-\uDD05\uDD07-\uDD0A\uDD0D-\uDD14\uDD16-\uDD1C\uDD1E-\uDD39\uDD3B-\uDD3E\uDD40-\uDD44\uDD46\uDD4A-\uDD50\uDD52-\uDEA5\uDEA8-\uDEC0\uDEC2-\uDEDA\uDEDC-\uDEFA\uDEFC-\uDF14\uDF16-\uDF34\uDF36-\uDF4E\uDF50-\uDF6E\uDF70-\uDF88\uDF8A-\uDFA8\uDFAA-\uDFC2\uDFC4-\uDFCB]|\uD83A[\uDC00-\uDCC4\uDD00-\uDD43]|\uD83B[\uDE00-\uDE03\uDE05-\uDE1F\uDE21\uDE22\uDE24\uDE27\uDE29-\uDE32\uDE34-\uDE37\uDE39\uDE3B\uDE42\uDE47\uDE49\uDE4B\uDE4D-\uDE4F\uDE51\uDE52\uDE54\uDE57\uDE59\uDE5B\uDE5D\uDE5F\uDE61\uDE62\uDE64\uDE67-\uDE6A\uDE6C-\uDE72\uDE74-\uDE77\uDE79-\uDE7C\uDE7E\uDE80-\uDE89\uDE8B-\uDE9B\uDEA1-\uDEA3\uDEA5-\uDEA9\uDEAB-\uDEBB]|\uD869[\uDC00-\uDED6\uDF00-\uDFFF]|\uD86D[\uDC00-\uDF34\uDF40-\uDFFF]|\uD86E[\uDC00-\uDC1D\uDC20-\uDFFF]|\uD873[\uDC00-\uDEA1\uDEB0-\uDFFF]|\uD87A[\uDC00-\uDFE0]|\uD87E[\uDC00-\uDE1D]/, O$1.ID_Continue = /[\xAA\xB5\xBA\xC0-\xD6\xD8-\xF6\xF8-\u02C1\u02C6-\u02D1\u02E0-\u02E4\u02EC\u02EE\u0300-\u0374\u0376\u0377\u037A-\u037D\u037F\u0386\u0388-\u038A\u038C\u038E-\u03A1\u03A3-\u03F5\u03F7-\u0481\u0483-\u0487\u048A-\u052F\u0531-\u0556\u0559\u0561-\u0587\u0591-\u05BD\u05BF\u05C1\u05C2\u05C4\u05C5\u05C7\u05D0-\u05EA\u05F0-\u05F2\u0610-\u061A\u0620-\u0669\u066E-\u06D3\u06D5-\u06DC\u06DF-\u06E8\u06EA-\u06FC\u06FF\u0710-\u074A\u074D-\u07B1\u07C0-\u07F5\u07FA\u0800-\u082D\u0840-\u085B\u0860-\u086A\u08A0-\u08B4\u08B6-\u08BD\u08D4-\u08E1\u08E3-\u0963\u0966-\u096F\u0971-\u0983\u0985-\u098C\u098F\u0990\u0993-\u09A8\u09AA-\u09B0\u09B2\u09B6-\u09B9\u09BC-\u09C4\u09C7\u09C8\u09CB-\u09CE\u09D7\u09DC\u09DD\u09DF-\u09E3\u09E6-\u09F1\u09FC\u0A01-\u0A03\u0A05-\u0A0A\u0A0F\u0A10\u0A13-\u0A28\u0A2A-\u0A30\u0A32\u0A33\u0A35\u0A36\u0A38\u0A39\u0A3C\u0A3E-\u0A42\u0A47\u0A48\u0A4B-\u0A4D\u0A51\u0A59-\u0A5C\u0A5E\u0A66-\u0A75\u0A81-\u0A83\u0A85-\u0A8D\u0A8F-\u0A91\u0A93-\u0AA8\u0AAA-\u0AB0\u0AB2\u0AB3\u0AB5-\u0AB9\u0ABC-\u0AC5\u0AC7-\u0AC9\u0ACB-\u0ACD\u0AD0\u0AE0-\u0AE3\u0AE6-\u0AEF\u0AF9-\u0AFF\u0B01-\u0B03\u0B05-\u0B0C\u0B0F\u0B10\u0B13-\u0B28\u0B2A-\u0B30\u0B32\u0B33\u0B35-\u0B39\u0B3C-\u0B44\u0B47\u0B48\u0B4B-\u0B4D\u0B56\u0B57\u0B5C\u0B5D\u0B5F-\u0B63\u0B66-\u0B6F\u0B71\u0B82\u0B83\u0B85-\u0B8A\u0B8E-\u0B90\u0B92-\u0B95\u0B99\u0B9A\u0B9C\u0B9E\u0B9F\u0BA3\u0BA4\u0BA8-\u0BAA\u0BAE-\u0BB9\u0BBE-\u0BC2\u0BC6-\u0BC8\u0BCA-\u0BCD\u0BD0\u0BD7\u0BE6-\u0BEF\u0C00-\u0C03\u0C05-\u0C0C\u0C0E-\u0C10\u0C12-\u0C28\u0C2A-\u0C39\u0C3D-\u0C44\u0C46-\u0C48\u0C4A-\u0C4D\u0C55\u0C56\u0C58-\u0C5A\u0C60-\u0C63\u0C66-\u0C6F\u0C80-\u0C83\u0C85-\u0C8C\u0C8E-\u0C90\u0C92-\u0CA8\u0CAA-\u0CB3\u0CB5-\u0CB9\u0CBC-\u0CC4\u0CC6-\u0CC8\u0CCA-\u0CCD\u0CD5\u0CD6\u0CDE\u0CE0-\u0CE3\u0CE6-\u0CEF\u0CF1\u0CF2\u0D00-\u0D03\u0D05-\u0D0C\u0D0E-\u0D10\u0D12-\u0D44\u0D46-\u0D48\u0D4A-\u0D4E\u0D54-\u0D57\u0D5F-\u0D63\u0D66-\u0D6F\u0D7A-\u0D7F\u0D82\u0D83\u0D85-\u0D96\u0D9A-\u0DB1\u0DB3-\u0DBB\u0DBD\u0DC0-\u0DC6\u0DCA\u0DCF-\u0DD4\u0DD6\u0DD8-\u0DDF\u0DE6-\u0DEF\u0DF2\u0DF3\u0E01-\u0E3A\u0E40-\u0E4E\u0E50-\u0E59\u0E81\u0E82\u0E84\u0E87\u0E88\u0E8A\u0E8D\u0E94-\u0E97\u0E99-\u0E9F\u0EA1-\u0EA3\u0EA5\u0EA7\u0EAA\u0EAB\u0EAD-\u0EB9\u0EBB-\u0EBD\u0EC0-\u0EC4\u0EC6\u0EC8-\u0ECD\u0ED0-\u0ED9\u0EDC-\u0EDF\u0F00\u0F18\u0F19\u0F20-\u0F29\u0F35\u0F37\u0F39\u0F3E-\u0F47\u0F49-\u0F6C\u0F71-\u0F84\u0F86-\u0F97\u0F99-\u0FBC\u0FC6\u1000-\u1049\u1050-\u109D\u10A0-\u10C5\u10C7\u10CD\u10D0-\u10FA\u10FC-\u1248\u124A-\u124D\u1250-\u1256\u1258\u125A-\u125D\u1260-\u1288\u128A-\u128D\u1290-\u12B0\u12B2-\u12B5\u12B8-\u12BE\u12C0\u12C2-\u12C5\u12C8-\u12D6\u12D8-\u1310\u1312-\u1315\u1318-\u135A\u135D-\u135F\u1380-\u138F\u13A0-\u13F5\u13F8-\u13FD\u1401-\u166C\u166F-\u167F\u1681-\u169A\u16A0-\u16EA\u16EE-\u16F8\u1700-\u170C\u170E-\u1714\u1720-\u1734\u1740-\u1753\u1760-\u176C\u176E-\u1770\u1772\u1773\u1780-\u17D3\u17D7\u17DC\u17DD\u17E0-\u17E9\u180B-\u180D\u1810-\u1819\u1820-\u1877\u1880-\u18AA\u18B0-\u18F5\u1900-\u191E\u1920-\u192B\u1930-\u193B\u1946-\u196D\u1970-\u1974\u1980-\u19AB\u19B0-\u19C9\u19D0-\u19D9\u1A00-\u1A1B\u1A20-\u1A5E\u1A60-\u1A7C\u1A7F-\u1A89\u1A90-\u1A99\u1AA7\u1AB0-\u1ABD\u1B00-\u1B4B\u1B50-\u1B59\u1B6B-\u1B73\u1B80-\u1BF3\u1C00-\u1C37\u1C40-\u1C49\u1C4D-\u1C7D\u1C80-\u1C88\u1CD0-\u1CD2\u1CD4-\u1CF9\u1D00-\u1DF9\u1DFB-\u1F15\u1F18-\u1F1D\u1F20-\u1F45\u1F48-\u1F4D\u1F50-\u1F57\u1F59\u1F5B\u1F5D\u1F5F-\u1F7D\u1F80-\u1FB4\u1FB6-\u1FBC\u1FBE\u1FC2-\u1FC4\u1FC6-\u1FCC\u1FD0-\u1FD3\u1FD6-\u1FDB\u1FE0-\u1FEC\u1FF2-\u1FF4\u1FF6-\u1FFC\u203F\u2040\u2054\u2071\u207F\u2090-\u209C\u20D0-\u20DC\u20E1\u20E5-\u20F0\u2102\u2107\u210A-\u2113\u2115\u2119-\u211D\u2124\u2126\u2128\u212A-\u212D\u212F-\u2139\u213C-\u213F\u2145-\u2149\u214E\u2160-\u2188\u2C00-\u2C2E\u2C30-\u2C5E\u2C60-\u2CE4\u2CEB-\u2CF3\u2D00-\u2D25\u2D27\u2D2D\u2D30-\u2D67\u2D6F\u2D7F-\u2D96\u2DA0-\u2DA6\u2DA8-\u2DAE\u2DB0-\u2DB6\u2DB8-\u2DBE\u2DC0-\u2DC6\u2DC8-\u2DCE\u2DD0-\u2DD6\u2DD8-\u2DDE\u2DE0-\u2DFF\u2E2F\u3005-\u3007\u3021-\u302F\u3031-\u3035\u3038-\u303C\u3041-\u3096\u3099\u309A\u309D-\u309F\u30A1-\u30FA\u30FC-\u30FF\u3105-\u312E\u3131-\u318E\u31A0-\u31BA\u31F0-\u31FF\u3400-\u4DB5\u4E00-\u9FEA\uA000-\uA48C\uA4D0-\uA4FD\uA500-\uA60C\uA610-\uA62B\uA640-\uA66F\uA674-\uA67D\uA67F-\uA6F1\uA717-\uA71F\uA722-\uA788\uA78B-\uA7AE\uA7B0-\uA7B7\uA7F7-\uA827\uA840-\uA873\uA880-\uA8C5\uA8D0-\uA8D9\uA8E0-\uA8F7\uA8FB\uA8FD\uA900-\uA92D\uA930-\uA953\uA960-\uA97C\uA980-\uA9C0\uA9CF-\uA9D9\uA9E0-\uA9FE\uAA00-\uAA36\uAA40-\uAA4D\uAA50-\uAA59\uAA60-\uAA76\uAA7A-\uAAC2\uAADB-\uAADD\uAAE0-\uAAEF\uAAF2-\uAAF6\uAB01-\uAB06\uAB09-\uAB0E\uAB11-\uAB16\uAB20-\uAB26\uAB28-\uAB2E\uAB30-\uAB5A\uAB5C-\uAB65\uAB70-\uABEA\uABEC\uABED\uABF0-\uABF9\uAC00-\uD7A3\uD7B0-\uD7C6\uD7CB-\uD7FB\uF900-\uFA6D\uFA70-\uFAD9\uFB00-\uFB06\uFB13-\uFB17\uFB1D-\uFB28\uFB2A-\uFB36\uFB38-\uFB3C\uFB3E\uFB40\uFB41\uFB43\uFB44\uFB46-\uFBB1\uFBD3-\uFD3D\uFD50-\uFD8F\uFD92-\uFDC7\uFDF0-\uFDFB\uFE00-\uFE0F\uFE20-\uFE2F\uFE33\uFE34\uFE4D-\uFE4F\uFE70-\uFE74\uFE76-\uFEFC\uFF10-\uFF19\uFF21-\uFF3A\uFF3F\uFF41-\uFF5A\uFF66-\uFFBE\uFFC2-\uFFC7\uFFCA-\uFFCF\uFFD2-\uFFD7\uFFDA-\uFFDC]|\uD800[\uDC00-\uDC0B\uDC0D-\uDC26\uDC28-\uDC3A\uDC3C\uDC3D\uDC3F-\uDC4D\uDC50-\uDC5D\uDC80-\uDCFA\uDD40-\uDD74\uDDFD\uDE80-\uDE9C\uDEA0-\uDED0\uDEE0\uDF00-\uDF1F\uDF2D-\uDF4A\uDF50-\uDF7A\uDF80-\uDF9D\uDFA0-\uDFC3\uDFC8-\uDFCF\uDFD1-\uDFD5]|\uD801[\uDC00-\uDC9D\uDCA0-\uDCA9\uDCB0-\uDCD3\uDCD8-\uDCFB\uDD00-\uDD27\uDD30-\uDD63\uDE00-\uDF36\uDF40-\uDF55\uDF60-\uDF67]|\uD802[\uDC00-\uDC05\uDC08\uDC0A-\uDC35\uDC37\uDC38\uDC3C\uDC3F-\uDC55\uDC60-\uDC76\uDC80-\uDC9E\uDCE0-\uDCF2\uDCF4\uDCF5\uDD00-\uDD15\uDD20-\uDD39\uDD80-\uDDB7\uDDBE\uDDBF\uDE00-\uDE03\uDE05\uDE06\uDE0C-\uDE13\uDE15-\uDE17\uDE19-\uDE33\uDE38-\uDE3A\uDE3F\uDE60-\uDE7C\uDE80-\uDE9C\uDEC0-\uDEC7\uDEC9-\uDEE6\uDF00-\uDF35\uDF40-\uDF55\uDF60-\uDF72\uDF80-\uDF91]|\uD803[\uDC00-\uDC48\uDC80-\uDCB2\uDCC0-\uDCF2]|\uD804[\uDC00-\uDC46\uDC66-\uDC6F\uDC7F-\uDCBA\uDCD0-\uDCE8\uDCF0-\uDCF9\uDD00-\uDD34\uDD36-\uDD3F\uDD50-\uDD73\uDD76\uDD80-\uDDC4\uDDCA-\uDDCC\uDDD0-\uDDDA\uDDDC\uDE00-\uDE11\uDE13-\uDE37\uDE3E\uDE80-\uDE86\uDE88\uDE8A-\uDE8D\uDE8F-\uDE9D\uDE9F-\uDEA8\uDEB0-\uDEEA\uDEF0-\uDEF9\uDF00-\uDF03\uDF05-\uDF0C\uDF0F\uDF10\uDF13-\uDF28\uDF2A-\uDF30\uDF32\uDF33\uDF35-\uDF39\uDF3C-\uDF44\uDF47\uDF48\uDF4B-\uDF4D\uDF50\uDF57\uDF5D-\uDF63\uDF66-\uDF6C\uDF70-\uDF74]|\uD805[\uDC00-\uDC4A\uDC50-\uDC59\uDC80-\uDCC5\uDCC7\uDCD0-\uDCD9\uDD80-\uDDB5\uDDB8-\uDDC0\uDDD8-\uDDDD\uDE00-\uDE40\uDE44\uDE50-\uDE59\uDE80-\uDEB7\uDEC0-\uDEC9\uDF00-\uDF19\uDF1D-\uDF2B\uDF30-\uDF39]|\uD806[\uDCA0-\uDCE9\uDCFF\uDE00-\uDE3E\uDE47\uDE50-\uDE83\uDE86-\uDE99\uDEC0-\uDEF8]|\uD807[\uDC00-\uDC08\uDC0A-\uDC36\uDC38-\uDC40\uDC50-\uDC59\uDC72-\uDC8F\uDC92-\uDCA7\uDCA9-\uDCB6\uDD00-\uDD06\uDD08\uDD09\uDD0B-\uDD36\uDD3A\uDD3C\uDD3D\uDD3F-\uDD47\uDD50-\uDD59]|\uD808[\uDC00-\uDF99]|\uD809[\uDC00-\uDC6E\uDC80-\uDD43]|[\uD80C\uD81C-\uD820\uD840-\uD868\uD86A-\uD86C\uD86F-\uD872\uD874-\uD879][\uDC00-\uDFFF]|\uD80D[\uDC00-\uDC2E]|\uD811[\uDC00-\uDE46]|\uD81A[\uDC00-\uDE38\uDE40-\uDE5E\uDE60-\uDE69\uDED0-\uDEED\uDEF0-\uDEF4\uDF00-\uDF36\uDF40-\uDF43\uDF50-\uDF59\uDF63-\uDF77\uDF7D-\uDF8F]|\uD81B[\uDF00-\uDF44\uDF50-\uDF7E\uDF8F-\uDF9F\uDFE0\uDFE1]|\uD821[\uDC00-\uDFEC]|\uD822[\uDC00-\uDEF2]|\uD82C[\uDC00-\uDD1E\uDD70-\uDEFB]|\uD82F[\uDC00-\uDC6A\uDC70-\uDC7C\uDC80-\uDC88\uDC90-\uDC99\uDC9D\uDC9E]|\uD834[\uDD65-\uDD69\uDD6D-\uDD72\uDD7B-\uDD82\uDD85-\uDD8B\uDDAA-\uDDAD\uDE42-\uDE44]|\uD835[\uDC00-\uDC54\uDC56-\uDC9C\uDC9E\uDC9F\uDCA2\uDCA5\uDCA6\uDCA9-\uDCAC\uDCAE-\uDCB9\uDCBB\uDCBD-\uDCC3\uDCC5-\uDD05\uDD07-\uDD0A\uDD0D-\uDD14\uDD16-\uDD1C\uDD1E-\uDD39\uDD3B-\uDD3E\uDD40-\uDD44\uDD46\uDD4A-\uDD50\uDD52-\uDEA5\uDEA8-\uDEC0\uDEC2-\uDEDA\uDEDC-\uDEFA\uDEFC-\uDF14\uDF16-\uDF34\uDF36-\uDF4E\uDF50-\uDF6E\uDF70-\uDF88\uDF8A-\uDFA8\uDFAA-\uDFC2\uDFC4-\uDFCB\uDFCE-\uDFFF]|\uD836[\uDE00-\uDE36\uDE3B-\uDE6C\uDE75\uDE84\uDE9B-\uDE9F\uDEA1-\uDEAF]|\uD838[\uDC00-\uDC06\uDC08-\uDC18\uDC1B-\uDC21\uDC23\uDC24\uDC26-\uDC2A]|\uD83A[\uDC00-\uDCC4\uDCD0-\uDCD6\uDD00-\uDD4A\uDD50-\uDD59]|\uD83B[\uDE00-\uDE03\uDE05-\uDE1F\uDE21\uDE22\uDE24\uDE27\uDE29-\uDE32\uDE34-\uDE37\uDE39\uDE3B\uDE42\uDE47\uDE49\uDE4B\uDE4D-\uDE4F\uDE51\uDE52\uDE54\uDE57\uDE59\uDE5B\uDE5D\uDE5F\uDE61\uDE62\uDE64\uDE67-\uDE6A\uDE6C-\uDE72\uDE74-\uDE77\uDE79-\uDE7C\uDE7E\uDE80-\uDE89\uDE8B-\uDE9B\uDEA1-\uDEA3\uDEA5-\uDEA9\uDEAB-\uDEBB]|\uD869[\uDC00-\uDED6\uDF00-\uDFFF]|\uD86D[\uDC00-\uDF34\uDF40-\uDFFF]|\uD86E[\uDC00-\uDC1D\uDC20-\uDFFF]|\uD873[\uDC00-\uDEA1\uDEB0-\uDFFF]|\uD87A[\uDC00-\uDFE0]|\uD87E[\uDC00-\uDE1D]|\uDB40[\uDD00-\uDDEF]/), O$1;
}
var _$1, U$2;
function T$1() {
	if (U$2) return _$1;
	U$2 = 1;
	const C$3 = Y$1();
	return _$1 = {
		isSpaceSeparator(r) {
			return typeof r == "string" && C$3.Space_Separator.test(r);
		},
		isIdStartChar(r) {
			return typeof r == "string" && (r >= "a" && r <= "z" || r >= "A" && r <= "Z" || r === "$" || r === "_" || C$3.ID_Start.test(r));
		},
		isIdContinueChar(r) {
			return typeof r == "string" && (r >= "a" && r <= "z" || r >= "A" && r <= "Z" || r >= "0" && r <= "9" || r === "$" || r === "_" || r === "‌" || r === "‍" || C$3.ID_Continue.test(r));
		},
		isDigit(r) {
			return typeof r == "string" && /[0-9]/.test(r);
		},
		isHexDigit(r) {
			return typeof r == "string" && /[0-9A-Fa-f]/.test(r);
		}
	}, _$1;
}
var q$1, Z$2;
function uu() {
	if (Z$2) return q$1;
	Z$2 = 1;
	const C$3 = T$1();
	let r, s$1, c$1, d, h$2, o, f, S$2, m;
	q$1 = function(a, g$1) {
		r = String(a), s$1 = "start", c$1 = [], d = 0, h$2 = 1, o = 0, f = void 0, S$2 = void 0, m = void 0;
		do
			f = E$1(), Q$2[s$1]();
		while (f.type !== "eof");
		return typeof g$1 == "function" ? v$2({ "": m }, "", g$1) : m;
	};
	function v$2(D$1, a, g$1) {
		const y$2 = D$1[a];
		if (y$2 != null && typeof y$2 == "object") if (Array.isArray(y$2)) for (let P$1 = 0; P$1 < y$2.length; P$1++) {
			const I$2 = String(P$1), H$1 = v$2(y$2, I$2, g$1);
			H$1 === void 0 ? delete y$2[I$2] : Object.defineProperty(y$2, I$2, {
				value: H$1,
				writable: !0,
				enumerable: !0,
				configurable: !0
			});
		}
		else for (const P$1 in y$2) {
			const I$2 = v$2(y$2, P$1, g$1);
			I$2 === void 0 ? delete y$2[P$1] : Object.defineProperty(y$2, P$1, {
				value: I$2,
				writable: !0,
				enumerable: !0,
				configurable: !0
			});
		}
		return g$1.call(D$1, a, y$2);
	}
	let t, e, x$1, w$2, A$1;
	function E$1() {
		for (t = "default", e = "", x$1 = !1, w$2 = 1;;) {
			A$1 = n();
			const D$1 = l[t]();
			if (D$1) return D$1;
		}
	}
	function n() {
		if (r[d]) return String.fromCodePoint(r.codePointAt(d));
	}
	function u() {
		const D$1 = n();
		return D$1 === `
` ? (h$2++, o = 0) : D$1 ? o += D$1.length : o++, D$1 && (d += D$1.length), D$1;
	}
	const l = {
		default() {
			switch (A$1) {
				case "	":
				case "\v":
				case "\f":
				case " ":
				case "\xA0":
				case "﻿":
				case `
`:
				case "\r":
				case "\u2028":
				case "\u2029":
					u();
					return;
				case "/":
					u(), t = "comment";
					return;
				case void 0: return u(), F$2("eof");
			}
			if (C$3.isSpaceSeparator(A$1)) {
				u();
				return;
			}
			return l[s$1]();
		},
		comment() {
			switch (A$1) {
				case "*":
					u(), t = "multiLineComment";
					return;
				case "/":
					u(), t = "singleLineComment";
					return;
			}
			throw B$1(u());
		},
		multiLineComment() {
			switch (A$1) {
				case "*":
					u(), t = "multiLineCommentAsterisk";
					return;
				case void 0: throw B$1(u());
			}
			u();
		},
		multiLineCommentAsterisk() {
			switch (A$1) {
				case "*":
					u();
					return;
				case "/":
					u(), t = "default";
					return;
				case void 0: throw B$1(u());
			}
			u(), t = "multiLineComment";
		},
		singleLineComment() {
			switch (A$1) {
				case `
`:
				case "\r":
				case "\u2028":
				case "\u2029":
					u(), t = "default";
					return;
				case void 0: return u(), F$2("eof");
			}
			u();
		},
		value() {
			switch (A$1) {
				case "{":
				case "[": return F$2("punctuator", u());
				case "n": return u(), i("ull"), F$2("null", null);
				case "t": return u(), i("rue"), F$2("boolean", !0);
				case "f": return u(), i("alse"), F$2("boolean", !1);
				case "-":
				case "+":
					u() === "-" && (w$2 = -1), t = "sign";
					return;
				case ".":
					e = u(), t = "decimalPointLeading";
					return;
				case "0":
					e = u(), t = "zero";
					return;
				case "1":
				case "2":
				case "3":
				case "4":
				case "5":
				case "6":
				case "7":
				case "8":
				case "9":
					e = u(), t = "decimalInteger";
					return;
				case "I": return u(), i("nfinity"), F$2("numeric", Infinity);
				case "N": return u(), i("aN"), F$2("numeric", NaN);
				case "\"":
				case "'":
					x$1 = u() === "\"", e = "", t = "string";
					return;
			}
			throw B$1(u());
		},
		identifierNameStartEscape() {
			if (A$1 !== "u") throw B$1(u());
			u();
			const D$1 = $$2();
			switch (D$1) {
				case "$":
				case "_": break;
				default:
					if (!C$3.isIdStartChar(D$1)) throw L$1();
					break;
			}
			e += D$1, t = "identifierName";
		},
		identifierName() {
			switch (A$1) {
				case "$":
				case "_":
				case "‌":
				case "‍":
					e += u();
					return;
				case "\\":
					u(), t = "identifierNameEscape";
					return;
			}
			if (C$3.isIdContinueChar(A$1)) {
				e += u();
				return;
			}
			return F$2("identifier", e);
		},
		identifierNameEscape() {
			if (A$1 !== "u") throw B$1(u());
			u();
			const D$1 = $$2();
			switch (D$1) {
				case "$":
				case "_":
				case "‌":
				case "‍": break;
				default:
					if (!C$3.isIdContinueChar(D$1)) throw L$1();
					break;
			}
			e += D$1, t = "identifierName";
		},
		sign() {
			switch (A$1) {
				case ".":
					e = u(), t = "decimalPointLeading";
					return;
				case "0":
					e = u(), t = "zero";
					return;
				case "1":
				case "2":
				case "3":
				case "4":
				case "5":
				case "6":
				case "7":
				case "8":
				case "9":
					e = u(), t = "decimalInteger";
					return;
				case "I": return u(), i("nfinity"), F$2("numeric", w$2 * Infinity);
				case "N": return u(), i("aN"), F$2("numeric", NaN);
			}
			throw B$1(u());
		},
		zero() {
			switch (A$1) {
				case ".":
					e += u(), t = "decimalPoint";
					return;
				case "e":
				case "E":
					e += u(), t = "decimalExponent";
					return;
				case "x":
				case "X":
					e += u(), t = "hexadecimal";
					return;
			}
			return F$2("numeric", w$2 * 0);
		},
		decimalInteger() {
			switch (A$1) {
				case ".":
					e += u(), t = "decimalPoint";
					return;
				case "e":
				case "E":
					e += u(), t = "decimalExponent";
					return;
			}
			if (C$3.isDigit(A$1)) {
				e += u();
				return;
			}
			return F$2("numeric", w$2 * Number(e));
		},
		decimalPointLeading() {
			if (C$3.isDigit(A$1)) {
				e += u(), t = "decimalFraction";
				return;
			}
			throw B$1(u());
		},
		decimalPoint() {
			switch (A$1) {
				case "e":
				case "E":
					e += u(), t = "decimalExponent";
					return;
			}
			if (C$3.isDigit(A$1)) {
				e += u(), t = "decimalFraction";
				return;
			}
			return F$2("numeric", w$2 * Number(e));
		},
		decimalFraction() {
			switch (A$1) {
				case "e":
				case "E":
					e += u(), t = "decimalExponent";
					return;
			}
			if (C$3.isDigit(A$1)) {
				e += u();
				return;
			}
			return F$2("numeric", w$2 * Number(e));
		},
		decimalExponent() {
			switch (A$1) {
				case "+":
				case "-":
					e += u(), t = "decimalExponentSign";
					return;
			}
			if (C$3.isDigit(A$1)) {
				e += u(), t = "decimalExponentInteger";
				return;
			}
			throw B$1(u());
		},
		decimalExponentSign() {
			if (C$3.isDigit(A$1)) {
				e += u(), t = "decimalExponentInteger";
				return;
			}
			throw B$1(u());
		},
		decimalExponentInteger() {
			if (C$3.isDigit(A$1)) {
				e += u();
				return;
			}
			return F$2("numeric", w$2 * Number(e));
		},
		hexadecimal() {
			if (C$3.isHexDigit(A$1)) {
				e += u(), t = "hexadecimalInteger";
				return;
			}
			throw B$1(u());
		},
		hexadecimalInteger() {
			if (C$3.isHexDigit(A$1)) {
				e += u();
				return;
			}
			return F$2("numeric", w$2 * Number(e));
		},
		string() {
			switch (A$1) {
				case "\\":
					u(), e += p$1();
					return;
				case "\"":
					if (x$1) return u(), F$2("string", e);
					e += u();
					return;
				case "'":
					if (!x$1) return u(), F$2("string", e);
					e += u();
					return;
				case `
`:
				case "\r": throw B$1(u());
				case "\u2028":
				case "\u2029":
					X$2(A$1);
					break;
				case void 0: throw B$1(u());
			}
			e += u();
		},
		start() {
			switch (A$1) {
				case "{":
				case "[": return F$2("punctuator", u());
			}
			t = "value";
		},
		beforePropertyName() {
			switch (A$1) {
				case "$":
				case "_":
					e = u(), t = "identifierName";
					return;
				case "\\":
					u(), t = "identifierNameStartEscape";
					return;
				case "}": return F$2("punctuator", u());
				case "\"":
				case "'":
					x$1 = u() === "\"", t = "string";
					return;
			}
			if (C$3.isIdStartChar(A$1)) {
				e += u(), t = "identifierName";
				return;
			}
			throw B$1(u());
		},
		afterPropertyName() {
			if (A$1 === ":") return F$2("punctuator", u());
			throw B$1(u());
		},
		beforePropertyValue() {
			t = "value";
		},
		afterPropertyValue() {
			switch (A$1) {
				case ",":
				case "}": return F$2("punctuator", u());
			}
			throw B$1(u());
		},
		beforeArrayValue() {
			if (A$1 === "]") return F$2("punctuator", u());
			t = "value";
		},
		afterArrayValue() {
			switch (A$1) {
				case ",":
				case "]": return F$2("punctuator", u());
			}
			throw B$1(u());
		},
		end() {
			throw B$1(u());
		}
	};
	function F$2(D$1, a) {
		return {
			type: D$1,
			value: a,
			line: h$2,
			column: o
		};
	}
	function i(D$1) {
		for (const a of D$1) {
			if (n() !== a) throw B$1(u());
			u();
		}
	}
	function p$1() {
		switch (n()) {
			case "b": return u(), "\b";
			case "f": return u(), "\f";
			case "n": return u(), `
`;
			case "r": return u(), "\r";
			case "t": return u(), "	";
			case "v": return u(), "\v";
			case "0":
				if (u(), C$3.isDigit(n())) throw B$1(u());
				return "\0";
			case "x": return u(), b$1();
			case "u": return u(), $$2();
			case `
`:
			case "\u2028":
			case "\u2029": return u(), "";
			case "\r": return u(), n() === `
` && u(), "";
			case "1":
			case "2":
			case "3":
			case "4":
			case "5":
			case "6":
			case "7":
			case "8":
			case "9": throw B$1(u());
			case void 0: throw B$1(u());
		}
		return u();
	}
	function b$1() {
		let D$1 = "", a = n();
		if (!C$3.isHexDigit(a) || (D$1 += u(), a = n(), !C$3.isHexDigit(a))) throw B$1(u());
		return D$1 += u(), String.fromCodePoint(parseInt(D$1, 16));
	}
	function $$2() {
		let D$1 = "", a = 4;
		for (; a-- > 0;) {
			const g$1 = n();
			if (!C$3.isHexDigit(g$1)) throw B$1(u());
			D$1 += u();
		}
		return String.fromCodePoint(parseInt(D$1, 16));
	}
	const Q$2 = {
		start() {
			if (f.type === "eof") throw N$2();
			V$2();
		},
		beforePropertyName() {
			switch (f.type) {
				case "identifier":
				case "string":
					S$2 = f.value, s$1 = "afterPropertyName";
					return;
				case "punctuator":
					j$2();
					return;
				case "eof": throw N$2();
			}
		},
		afterPropertyName() {
			if (f.type === "eof") throw N$2();
			s$1 = "beforePropertyValue";
		},
		beforePropertyValue() {
			if (f.type === "eof") throw N$2();
			V$2();
		},
		beforeArrayValue() {
			if (f.type === "eof") throw N$2();
			if (f.type === "punctuator" && f.value === "]") {
				j$2();
				return;
			}
			V$2();
		},
		afterPropertyValue() {
			if (f.type === "eof") throw N$2();
			switch (f.value) {
				case ",":
					s$1 = "beforePropertyName";
					return;
				case "}": j$2();
			}
		},
		afterArrayValue() {
			if (f.type === "eof") throw N$2();
			switch (f.value) {
				case ",":
					s$1 = "beforeArrayValue";
					return;
				case "]": j$2();
			}
		},
		end() {}
	};
	function V$2() {
		let D$1;
		switch (f.type) {
			case "punctuator":
				switch (f.value) {
					case "{":
						D$1 = {};
						break;
					case "[":
						D$1 = [];
						break;
				}
				break;
			case "null":
			case "boolean":
			case "numeric":
			case "string":
				D$1 = f.value;
				break;
		}
		if (m === void 0) m = D$1;
		else {
			const a = c$1[c$1.length - 1];
			Array.isArray(a) ? a.push(D$1) : Object.defineProperty(a, S$2, {
				value: D$1,
				writable: !0,
				enumerable: !0,
				configurable: !0
			});
		}
		if (D$1 !== null && typeof D$1 == "object") c$1.push(D$1), Array.isArray(D$1) ? s$1 = "beforeArrayValue" : s$1 = "beforePropertyName";
		else {
			const a = c$1[c$1.length - 1];
			a == null ? s$1 = "end" : Array.isArray(a) ? s$1 = "afterArrayValue" : s$1 = "afterPropertyValue";
		}
	}
	function j$2() {
		c$1.pop();
		const D$1 = c$1[c$1.length - 1];
		D$1 == null ? s$1 = "end" : Array.isArray(D$1) ? s$1 = "afterArrayValue" : s$1 = "afterPropertyValue";
	}
	function B$1(D$1) {
		return k$1(D$1 === void 0 ? `JSON5: invalid end of input at ${h$2}:${o}` : `JSON5: invalid character '${z$2(D$1)}' at ${h$2}:${o}`);
	}
	function N$2() {
		return k$1(`JSON5: invalid end of input at ${h$2}:${o}`);
	}
	function L$1() {
		return o -= 5, k$1(`JSON5: invalid identifier character at ${h$2}:${o}`);
	}
	function X$2(D$1) {
		console.warn(`JSON5: '${z$2(D$1)}' in strings is not valid ECMAScript; consider escaping`);
	}
	function z$2(D$1) {
		const a = {
			"'": "\\'",
			"\"": "\\\"",
			"\\": "\\\\",
			"\b": "\\b",
			"\f": "\\f",
			"\n": "\\n",
			"\r": "\\r",
			"	": "\\t",
			"\v": "\\v",
			"\0": "\\0",
			"\u2028": "\\u2028",
			"\u2029": "\\u2029"
		};
		if (a[D$1]) return a[D$1];
		if (D$1 < " ") {
			const g$1 = D$1.charCodeAt(0).toString(16);
			return "\\x" + ("00" + g$1).substring(g$1.length);
		}
		return D$1;
	}
	function k$1(D$1) {
		const a = new SyntaxError(D$1);
		return a.lineNumber = h$2, a.columnNumber = o, a;
	}
	return q$1;
}
const eu = R$2(uu());
var J$1, K$2;
function tu() {
	if (K$2) return J$1;
	K$2 = 1;
	const C$3 = T$1();
	return J$1 = function(s$1, c$1, d) {
		const h$2 = [];
		let o = "", f, S$2, m = "", v$2;
		if (c$1 != null && typeof c$1 == "object" && !Array.isArray(c$1) && (d = c$1.space, v$2 = c$1.quote, c$1 = c$1.replacer), typeof c$1 == "function") S$2 = c$1;
		else if (Array.isArray(c$1)) {
			f = [];
			for (const E$1 of c$1) {
				let n;
				typeof E$1 == "string" ? n = E$1 : (typeof E$1 == "number" || E$1 instanceof String || E$1 instanceof Number) && (n = String(E$1)), n !== void 0 && f.indexOf(n) < 0 && f.push(n);
			}
		}
		return d instanceof Number ? d = Number(d) : d instanceof String && (d = String(d)), typeof d == "number" ? d > 0 && (d = Math.min(10, Math.floor(d)), m = "          ".substr(0, d)) : typeof d == "string" && (m = d.substr(0, 10)), t("", { "": s$1 });
		function t(E$1, n) {
			let u = n[E$1];
			switch (u != null && (typeof u.toJSON5 == "function" ? u = u.toJSON5(E$1) : typeof u.toJSON == "function" && (u = u.toJSON(E$1))), S$2 && (u = S$2.call(n, E$1, u)), u instanceof Number ? u = Number(u) : u instanceof String ? u = String(u) : u instanceof Boolean && (u = u.valueOf()), u) {
				case null: return "null";
				case !0: return "true";
				case !1: return "false";
			}
			if (typeof u == "string") return e(u);
			if (typeof u == "number") return String(u);
			if (typeof u == "object") return Array.isArray(u) ? A$1(u) : x$1(u);
		}
		function e(E$1) {
			const n = {
				"'": .1,
				"\"": .2
			}, u = {
				"'": "\\'",
				"\"": "\\\"",
				"\\": "\\\\",
				"\b": "\\b",
				"\f": "\\f",
				"\n": "\\n",
				"\r": "\\r",
				"	": "\\t",
				"\v": "\\v",
				"\0": "\\0",
				"\u2028": "\\u2028",
				"\u2029": "\\u2029"
			};
			let l = "";
			for (let i = 0; i < E$1.length; i++) {
				const p$1 = E$1[i];
				switch (p$1) {
					case "'":
					case "\"":
						n[p$1]++, l += p$1;
						continue;
					case "\0": if (C$3.isDigit(E$1[i + 1])) {
						l += "\\x00";
						continue;
					}
				}
				if (u[p$1]) {
					l += u[p$1];
					continue;
				}
				if (p$1 < " ") {
					let b$1 = p$1.charCodeAt(0).toString(16);
					l += "\\x" + ("00" + b$1).substring(b$1.length);
					continue;
				}
				l += p$1;
			}
			const F$2 = v$2 || Object.keys(n).reduce((i, p$1) => n[i] < n[p$1] ? i : p$1);
			return l = l.replace(new RegExp(F$2, "g"), u[F$2]), F$2 + l + F$2;
		}
		function x$1(E$1) {
			if (h$2.indexOf(E$1) >= 0) throw TypeError("Converting circular structure to JSON5");
			h$2.push(E$1);
			let n = o;
			o = o + m;
			let u = f || Object.keys(E$1), l = [];
			for (const i of u) {
				const p$1 = t(i, E$1);
				if (p$1 !== void 0) {
					let b$1 = w$2(i) + ":";
					m !== "" && (b$1 += " "), b$1 += p$1, l.push(b$1);
				}
			}
			let F$2;
			if (l.length === 0) F$2 = "{}";
			else {
				let i;
				if (m === "") i = l.join(","), F$2 = "{" + i + "}";
				else {
					let p$1 = `,
` + o;
					i = l.join(p$1), F$2 = `{
` + o + i + `,
` + n + "}";
				}
			}
			return h$2.pop(), o = n, F$2;
		}
		function w$2(E$1) {
			if (E$1.length === 0) return e(E$1);
			const n = String.fromCodePoint(E$1.codePointAt(0));
			if (!C$3.isIdStartChar(n)) return e(E$1);
			for (let u = n.length; u < E$1.length; u++) if (!C$3.isIdContinueChar(String.fromCodePoint(E$1.codePointAt(u)))) return e(E$1);
			return E$1;
		}
		function A$1(E$1) {
			if (h$2.indexOf(E$1) >= 0) throw TypeError("Converting circular structure to JSON5");
			h$2.push(E$1);
			let n = o;
			o = o + m;
			let u = [];
			for (let F$2 = 0; F$2 < E$1.length; F$2++) {
				const i = t(String(F$2), E$1);
				u.push(i !== void 0 ? i : "null");
			}
			let l;
			if (u.length === 0) l = "[]";
			else if (m === "") l = "[" + u.join(",") + "]";
			else {
				let F$2 = `,
` + o, i = u.join(F$2);
				l = `[
` + o + i + `,
` + n + "]";
			}
			return h$2.pop(), o = n, l;
		}
	}, J$1;
}
const ru = R$2(tu());
function Cu(C$3, r) {
	const s$1 = eu(C$3, r?.reviver);
	return N$1(C$3, s$1, r), s$1;
}

//#endregion
//#region node_modules/.pnpm/confbox@0.2.2/node_modules/confbox/dist/yaml.mjs
var yaml_exports = /* @__PURE__ */ __exportAll({
	parseYAML: () => mr,
	stringifyYAML: () => gr
});
/*! js-yaml 4.1.0 https://github.com/nodeca/js-yaml @license MIT */ function oe(e) {
	return typeof e > "u" || e === null;
}
function Ge(e) {
	return typeof e == "object" && e !== null;
}
function We(e) {
	return Array.isArray(e) ? e : oe(e) ? [] : [e];
}
function $e(e, n) {
	var i, l, r, u;
	if (n) for (u = Object.keys(n), i = 0, l = u.length; i < l; i += 1) r = u[i], e[r] = n[r];
	return e;
}
function Qe(e, n) {
	var i = "", l;
	for (l = 0; l < n; l += 1) i += e;
	return i;
}
function Ve(e) {
	return e === 0 && Number.NEGATIVE_INFINITY === 1 / e;
}
var y$1 = {
	isNothing: oe,
	isObject: Ge,
	toArray: We,
	repeat: Qe,
	isNegativeZero: Ve,
	extend: $e
};
function ue(e, n) {
	var i = "", l = e.reason || "(unknown reason)";
	return e.mark ? (e.mark.name && (i += "in \"" + e.mark.name + "\" "), i += "(" + (e.mark.line + 1) + ":" + (e.mark.column + 1) + ")", !n && e.mark.snippet && (i += `

` + e.mark.snippet), l + " " + i) : l;
}
function M$1(e, n) {
	Error.call(this), this.name = "YAMLException", this.reason = e, this.mark = n, this.message = ue(this, !1), Error.captureStackTrace ? Error.captureStackTrace(this, this.constructor) : this.stack = (/* @__PURE__ */ new Error()).stack || "";
}
M$1.prototype = Object.create(Error.prototype), M$1.prototype.constructor = M$1, M$1.prototype.toString = function(n) {
	return this.name + ": " + ue(this, n);
};
var w$1 = M$1;
function $$1(e, n, i, l, r) {
	var u = "", o = "", f = Math.floor(r / 2) - 1;
	return l - n > f && (u = " ... ", n = l - f + u.length), i - l > f && (o = " ...", i = l + f - o.length), {
		str: u + e.slice(n, i).replace(/\t/g, "→") + o,
		pos: l - n + u.length
	};
}
function Q$1(e, n) {
	return y$1.repeat(" ", n - e.length) + e;
}
function rn(e, n) {
	if (n = Object.create(n || null), !e.buffer) return null;
	n.maxLength || (n.maxLength = 79), typeof n.indent != "number" && (n.indent = 1), typeof n.linesBefore != "number" && (n.linesBefore = 3), typeof n.linesAfter != "number" && (n.linesAfter = 2);
	for (var i = /\r?\n|\r|\0/g, l = [0], r = [], u, o = -1; u = i.exec(e.buffer);) r.push(u.index), l.push(u.index + u[0].length), e.position <= u.index && o < 0 && (o = l.length - 2);
	o < 0 && (o = l.length - 1);
	var f = "", c$1, a, t = Math.min(e.line + n.linesAfter, r.length).toString().length, p$1 = n.maxLength - (n.indent + t + 3);
	for (c$1 = 1; c$1 <= n.linesBefore && !(o - c$1 < 0); c$1++) a = $$1(e.buffer, l[o - c$1], r[o - c$1], e.position - (l[o] - l[o - c$1]), p$1), f = y$1.repeat(" ", n.indent) + Q$1((e.line - c$1 + 1).toString(), t) + " | " + a.str + `
` + f;
	for (a = $$1(e.buffer, l[o], r[o], e.position, p$1), f += y$1.repeat(" ", n.indent) + Q$1((e.line + 1).toString(), t) + " | " + a.str + `
`, f += y$1.repeat("-", n.indent + t + 3 + a.pos) + `^
`, c$1 = 1; c$1 <= n.linesAfter && !(o + c$1 >= r.length); c$1++) a = $$1(e.buffer, l[o + c$1], r[o + c$1], e.position - (l[o] - l[o + c$1]), p$1), f += y$1.repeat(" ", n.indent) + Q$1((e.line + c$1 + 1).toString(), t) + " | " + a.str + `
`;
	return f.replace(/\n$/, "");
}
var ln = rn, on = [
	"kind",
	"multi",
	"resolve",
	"construct",
	"instanceOf",
	"predicate",
	"represent",
	"representName",
	"defaultStyle",
	"styleAliases"
], un = [
	"scalar",
	"sequence",
	"mapping"
];
function fn(e) {
	var n = {};
	return e !== null && Object.keys(e).forEach(function(i) {
		e[i].forEach(function(l) {
			n[String(l)] = i;
		});
	}), n;
}
function cn(e, n) {
	if (n = n || {}, Object.keys(n).forEach(function(i) {
		if (on.indexOf(i) === -1) throw new w$1("Unknown option \"" + i + "\" is met in definition of \"" + e + "\" YAML type.");
	}), this.options = n, this.tag = e, this.kind = n.kind || null, this.resolve = n.resolve || function() {
		return !0;
	}, this.construct = n.construct || function(i) {
		return i;
	}, this.instanceOf = n.instanceOf || null, this.predicate = n.predicate || null, this.represent = n.represent || null, this.representName = n.representName || null, this.defaultStyle = n.defaultStyle || null, this.multi = n.multi || !1, this.styleAliases = fn(n.styleAliases || null), un.indexOf(this.kind) === -1) throw new w$1("Unknown kind \"" + this.kind + "\" is specified for \"" + e + "\" YAML type.");
}
var C$1 = cn;
function fe(e, n) {
	var i = [];
	return e[n].forEach(function(l) {
		var r = i.length;
		i.forEach(function(u, o) {
			u.tag === l.tag && u.kind === l.kind && u.multi === l.multi && (r = o);
		}), i[r] = l;
	}), i;
}
function an() {
	var e = {
		scalar: {},
		sequence: {},
		mapping: {},
		fallback: {},
		multi: {
			scalar: [],
			sequence: [],
			mapping: [],
			fallback: []
		}
	}, n, i;
	function l(r) {
		r.multi ? (e.multi[r.kind].push(r), e.multi.fallback.push(r)) : e[r.kind][r.tag] = e.fallback[r.tag] = r;
	}
	for (n = 0, i = arguments.length; n < i; n += 1) arguments[n].forEach(l);
	return e;
}
function V$1(e) {
	return this.extend(e);
}
V$1.prototype.extend = function(n) {
	var i = [], l = [];
	if (n instanceof C$1) l.push(n);
	else if (Array.isArray(n)) l = l.concat(n);
	else if (n && (Array.isArray(n.implicit) || Array.isArray(n.explicit))) n.implicit && (i = i.concat(n.implicit)), n.explicit && (l = l.concat(n.explicit));
	else throw new w$1("Schema.extend argument should be a Type, [ Type ], or a schema definition ({ implicit: [...], explicit: [...] })");
	i.forEach(function(u) {
		if (!(u instanceof C$1)) throw new w$1("Specified list of YAML types (or a single Type object) contains a non-Type object.");
		if (u.loadKind && u.loadKind !== "scalar") throw new w$1("There is a non-scalar type in the implicit list of a schema. Implicit resolving of such types is not supported.");
		if (u.multi) throw new w$1("There is a multi type in the implicit list of a schema. Multi tags can only be listed as explicit.");
	}), l.forEach(function(u) {
		if (!(u instanceof C$1)) throw new w$1("Specified list of YAML types (or a single Type object) contains a non-Type object.");
	});
	var r = Object.create(V$1.prototype);
	return r.implicit = (this.implicit || []).concat(i), r.explicit = (this.explicit || []).concat(l), r.compiledImplicit = fe(r, "implicit"), r.compiledExplicit = fe(r, "explicit"), r.compiledTypeMap = an(r.compiledImplicit, r.compiledExplicit), r;
};
var sn = new V$1({ explicit: [
	new C$1("tag:yaml.org,2002:str", {
		kind: "scalar",
		construct: function(e) {
			return e !== null ? e : "";
		}
	}),
	new C$1("tag:yaml.org,2002:seq", {
		kind: "sequence",
		construct: function(e) {
			return e !== null ? e : [];
		}
	}),
	new C$1("tag:yaml.org,2002:map", {
		kind: "mapping",
		construct: function(e) {
			return e !== null ? e : {};
		}
	})
] });
function xn(e) {
	if (e === null) return !0;
	var n = e.length;
	return n === 1 && e === "~" || n === 4 && (e === "null" || e === "Null" || e === "NULL");
}
function mn() {
	return null;
}
function gn(e) {
	return e === null;
}
var An = new C$1("tag:yaml.org,2002:null", {
	kind: "scalar",
	resolve: xn,
	construct: mn,
	predicate: gn,
	represent: {
		canonical: function() {
			return "~";
		},
		lowercase: function() {
			return "null";
		},
		uppercase: function() {
			return "NULL";
		},
		camelcase: function() {
			return "Null";
		},
		empty: function() {
			return "";
		}
	},
	defaultStyle: "lowercase"
});
function vn(e) {
	if (e === null) return !1;
	var n = e.length;
	return n === 4 && (e === "true" || e === "True" || e === "TRUE") || n === 5 && (e === "false" || e === "False" || e === "FALSE");
}
function yn(e) {
	return e === "true" || e === "True" || e === "TRUE";
}
function Cn(e) {
	return Object.prototype.toString.call(e) === "[object Boolean]";
}
var _n = new C$1("tag:yaml.org,2002:bool", {
	kind: "scalar",
	resolve: vn,
	construct: yn,
	predicate: Cn,
	represent: {
		lowercase: function(e) {
			return e ? "true" : "false";
		},
		uppercase: function(e) {
			return e ? "TRUE" : "FALSE";
		},
		camelcase: function(e) {
			return e ? "True" : "False";
		}
	},
	defaultStyle: "lowercase"
});
function wn(e) {
	return 48 <= e && e <= 57 || 65 <= e && e <= 70 || 97 <= e && e <= 102;
}
function Fn(e) {
	return 48 <= e && e <= 55;
}
function bn(e) {
	return 48 <= e && e <= 57;
}
function Sn(e) {
	if (e === null) return !1;
	var n = e.length, i = 0, l = !1, r;
	if (!n) return !1;
	if (r = e[i], (r === "-" || r === "+") && (r = e[++i]), r === "0") {
		if (i + 1 === n) return !0;
		if (r = e[++i], r === "b") {
			for (i++; i < n; i++) if (r = e[i], r !== "_") {
				if (r !== "0" && r !== "1") return !1;
				l = !0;
			}
			return l && r !== "_";
		}
		if (r === "x") {
			for (i++; i < n; i++) if (r = e[i], r !== "_") {
				if (!wn(e.charCodeAt(i))) return !1;
				l = !0;
			}
			return l && r !== "_";
		}
		if (r === "o") {
			for (i++; i < n; i++) if (r = e[i], r !== "_") {
				if (!Fn(e.charCodeAt(i))) return !1;
				l = !0;
			}
			return l && r !== "_";
		}
	}
	if (r === "_") return !1;
	for (; i < n; i++) if (r = e[i], r !== "_") {
		if (!bn(e.charCodeAt(i))) return !1;
		l = !0;
	}
	return !(!l || r === "_");
}
function En(e) {
	var n = e, i = 1, l;
	if (n.indexOf("_") !== -1 && (n = n.replace(/_/g, "")), l = n[0], (l === "-" || l === "+") && (l === "-" && (i = -1), n = n.slice(1), l = n[0]), n === "0") return 0;
	if (l === "0") {
		if (n[1] === "b") return i * parseInt(n.slice(2), 2);
		if (n[1] === "x") return i * parseInt(n.slice(2), 16);
		if (n[1] === "o") return i * parseInt(n.slice(2), 8);
	}
	return i * parseInt(n, 10);
}
function Tn(e) {
	return Object.prototype.toString.call(e) === "[object Number]" && e % 1 === 0 && !y$1.isNegativeZero(e);
}
var On = new C$1("tag:yaml.org,2002:int", {
	kind: "scalar",
	resolve: Sn,
	construct: En,
	predicate: Tn,
	represent: {
		binary: function(e) {
			return e >= 0 ? "0b" + e.toString(2) : "-0b" + e.toString(2).slice(1);
		},
		octal: function(e) {
			return e >= 0 ? "0o" + e.toString(8) : "-0o" + e.toString(8).slice(1);
		},
		decimal: function(e) {
			return e.toString(10);
		},
		hexadecimal: function(e) {
			return e >= 0 ? "0x" + e.toString(16).toUpperCase() : "-0x" + e.toString(16).toUpperCase().slice(1);
		}
	},
	defaultStyle: "decimal",
	styleAliases: {
		binary: [2, "bin"],
		octal: [8, "oct"],
		decimal: [10, "dec"],
		hexadecimal: [16, "hex"]
	}
}), In = /* @__PURE__ */ new RegExp("^(?:[-+]?(?:[0-9][0-9_]*)(?:\\.[0-9_]*)?(?:[eE][-+]?[0-9]+)?|\\.[0-9_]+(?:[eE][-+]?[0-9]+)?|[-+]?\\.(?:inf|Inf|INF)|\\.(?:nan|NaN|NAN))$");
function kn(e) {
	return !(e === null || !In.test(e) || e[e.length - 1] === "_");
}
function Ln(e) {
	var n, i;
	return n = e.replace(/_/g, "").toLowerCase(), i = n[0] === "-" ? -1 : 1, "+-".indexOf(n[0]) >= 0 && (n = n.slice(1)), n === ".inf" ? i === 1 ? Number.POSITIVE_INFINITY : Number.NEGATIVE_INFINITY : n === ".nan" ? NaN : i * parseFloat(n, 10);
}
var Nn = /^[-+]?[0-9]+e/;
function Rn(e, n) {
	var i;
	if (isNaN(e)) switch (n) {
		case "lowercase": return ".nan";
		case "uppercase": return ".NAN";
		case "camelcase": return ".NaN";
	}
	else if (Number.POSITIVE_INFINITY === e) switch (n) {
		case "lowercase": return ".inf";
		case "uppercase": return ".INF";
		case "camelcase": return ".Inf";
	}
	else if (Number.NEGATIVE_INFINITY === e) switch (n) {
		case "lowercase": return "-.inf";
		case "uppercase": return "-.INF";
		case "camelcase": return "-.Inf";
	}
	else if (y$1.isNegativeZero(e)) return "-0.0";
	return i = e.toString(10), Nn.test(i) ? i.replace("e", ".e") : i;
}
function Dn(e) {
	return Object.prototype.toString.call(e) === "[object Number]" && (e % 1 !== 0 || y$1.isNegativeZero(e));
}
var Mn = new C$1("tag:yaml.org,2002:float", {
	kind: "scalar",
	resolve: kn,
	construct: Ln,
	predicate: Dn,
	represent: Rn,
	defaultStyle: "lowercase"
}), Bn = sn.extend({ implicit: [
	An,
	_n,
	On,
	Mn
] }), ce = /* @__PURE__ */ new RegExp("^([0-9][0-9][0-9][0-9])-([0-9][0-9])-([0-9][0-9])$"), ae = /* @__PURE__ */ new RegExp("^([0-9][0-9][0-9][0-9])-([0-9][0-9]?)-([0-9][0-9]?)(?:[Tt]|[ \\t]+)([0-9][0-9]?):([0-9][0-9]):([0-9][0-9])(?:\\.([0-9]*))?(?:[ \\t]*(Z|([-+])([0-9][0-9]?)(?::([0-9][0-9]))?))?$");
function Pn(e) {
	return e === null ? !1 : ce.exec(e) !== null || ae.exec(e) !== null;
}
function jn(e) {
	var n, i, l, r, u, o, f, c$1 = 0, a = null, t, p$1, d;
	if (n = ce.exec(e), n === null && (n = ae.exec(e)), n === null) throw new Error("Date resolve error");
	if (i = +n[1], l = +n[2] - 1, r = +n[3], !n[4]) return new Date(Date.UTC(i, l, r));
	if (u = +n[4], o = +n[5], f = +n[6], n[7]) {
		for (c$1 = n[7].slice(0, 3); c$1.length < 3;) c$1 += "0";
		c$1 = +c$1;
	}
	return n[9] && (t = +n[10], p$1 = +(n[11] || 0), a = (t * 60 + p$1) * 6e4, n[9] === "-" && (a = -a)), d = new Date(Date.UTC(i, l, r, u, o, f, c$1)), a && d.setTime(d.getTime() - a), d;
}
function Hn(e) {
	return e.toISOString();
}
var Un = new C$1("tag:yaml.org,2002:timestamp", {
	kind: "scalar",
	resolve: Pn,
	construct: jn,
	instanceOf: Date,
	represent: Hn
});
function Kn(e) {
	return e === "<<" || e === null;
}
var qn = new C$1("tag:yaml.org,2002:merge", {
	kind: "scalar",
	resolve: Kn
}), X$1 = `ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=
\r`;
function Gn(e) {
	if (e === null) return !1;
	var n, i, l = 0, r = e.length, u = X$1;
	for (i = 0; i < r; i++) if (n = u.indexOf(e.charAt(i)), !(n > 64)) {
		if (n < 0) return !1;
		l += 6;
	}
	return l % 8 === 0;
}
function Wn(e) {
	var n, i, l = e.replace(/[\r\n=]/g, ""), r = l.length, u = X$1, o = 0, f = [];
	for (n = 0; n < r; n++) n % 4 === 0 && n && (f.push(o >> 16 & 255), f.push(o >> 8 & 255), f.push(o & 255)), o = o << 6 | u.indexOf(l.charAt(n));
	return i = r % 4 * 6, i === 0 ? (f.push(o >> 16 & 255), f.push(o >> 8 & 255), f.push(o & 255)) : i === 18 ? (f.push(o >> 10 & 255), f.push(o >> 2 & 255)) : i === 12 && f.push(o >> 4 & 255), new Uint8Array(f);
}
function $n(e) {
	var n = "", i = 0, l, r, u = e.length, o = X$1;
	for (l = 0; l < u; l++) l % 3 === 0 && l && (n += o[i >> 18 & 63], n += o[i >> 12 & 63], n += o[i >> 6 & 63], n += o[i & 63]), i = (i << 8) + e[l];
	return r = u % 3, r === 0 ? (n += o[i >> 18 & 63], n += o[i >> 12 & 63], n += o[i >> 6 & 63], n += o[i & 63]) : r === 2 ? (n += o[i >> 10 & 63], n += o[i >> 4 & 63], n += o[i << 2 & 63], n += o[64]) : r === 1 && (n += o[i >> 2 & 63], n += o[i << 4 & 63], n += o[64], n += o[64]), n;
}
function Qn(e) {
	return Object.prototype.toString.call(e) === "[object Uint8Array]";
}
var Vn = new C$1("tag:yaml.org,2002:binary", {
	kind: "scalar",
	resolve: Gn,
	construct: Wn,
	predicate: Qn,
	represent: $n
}), Xn = Object.prototype.hasOwnProperty, Zn = Object.prototype.toString;
function zn(e) {
	if (e === null) return !0;
	var n = [], i, l, r, u, o, f = e;
	for (i = 0, l = f.length; i < l; i += 1) {
		if (r = f[i], o = !1, Zn.call(r) !== "[object Object]") return !1;
		for (u in r) if (Xn.call(r, u)) if (!o) o = !0;
		else return !1;
		if (!o) return !1;
		if (n.indexOf(u) === -1) n.push(u);
		else return !1;
	}
	return !0;
}
function Jn(e) {
	return e !== null ? e : [];
}
var ei = new C$1("tag:yaml.org,2002:omap", {
	kind: "sequence",
	resolve: zn,
	construct: Jn
}), ni = Object.prototype.toString;
function ii(e) {
	if (e === null) return !0;
	var n, i, l, r, u, o = e;
	for (u = new Array(o.length), n = 0, i = o.length; n < i; n += 1) {
		if (l = o[n], ni.call(l) !== "[object Object]" || (r = Object.keys(l), r.length !== 1)) return !1;
		u[n] = [r[0], l[r[0]]];
	}
	return !0;
}
function ri(e) {
	if (e === null) return [];
	var n, i, l, r, u, o = e;
	for (u = new Array(o.length), n = 0, i = o.length; n < i; n += 1) l = o[n], r = Object.keys(l), u[n] = [r[0], l[r[0]]];
	return u;
}
var li = new C$1("tag:yaml.org,2002:pairs", {
	kind: "sequence",
	resolve: ii,
	construct: ri
}), oi = Object.prototype.hasOwnProperty;
function ui(e) {
	if (e === null) return !0;
	var n, i = e;
	for (n in i) if (oi.call(i, n) && i[n] !== null) return !1;
	return !0;
}
function fi(e) {
	return e !== null ? e : {};
}
var ci = new C$1("tag:yaml.org,2002:set", {
	kind: "mapping",
	resolve: ui,
	construct: fi
}), pe = Bn.extend({
	implicit: [Un, qn],
	explicit: [
		Vn,
		ei,
		li,
		ci
	]
}), T = Object.prototype.hasOwnProperty, H = 1, te = 2, he = 3, U$1 = 4, Z$1 = 1, ai = 2, de = 3, pi = /[\x00-\x08\x0B\x0C\x0E-\x1F\x7F-\x84\x86-\x9F\uFFFE\uFFFF]|[\uD800-\uDBFF](?![\uDC00-\uDFFF])|(?:[^\uD800-\uDBFF]|^)[\uDC00-\uDFFF]/, ti = /[\x85\u2028\u2029]/, hi = /[,\[\]\{\}]/, se = /^(?:!|!!|![a-z\-]+!)$/i, xe = /^(?:!|[^,\[\]\{\}])(?:%[0-9a-f]{2}|[0-9a-z\-#;\/\?:@&=\+\$,_\.!~\*'\(\)\[\]])*$/i;
function me(e) {
	return Object.prototype.toString.call(e);
}
function S$1(e) {
	return e === 10 || e === 13;
}
function I$1(e) {
	return e === 9 || e === 32;
}
function F$1(e) {
	return e === 9 || e === 32 || e === 10 || e === 13;
}
function k(e) {
	return e === 44 || e === 91 || e === 93 || e === 123 || e === 125;
}
function di(e) {
	var n;
	return 48 <= e && e <= 57 ? e - 48 : (n = e | 32, 97 <= n && n <= 102 ? n - 97 + 10 : -1);
}
function si(e) {
	return e === 120 ? 2 : e === 117 ? 4 : e === 85 ? 8 : 0;
}
function xi(e) {
	return 48 <= e && e <= 57 ? e - 48 : -1;
}
function ge(e) {
	return e === 48 ? "\0" : e === 97 ? "\x07" : e === 98 ? "\b" : e === 116 || e === 9 ? "	" : e === 110 ? `
` : e === 118 ? "\v" : e === 102 ? "\f" : e === 114 ? "\r" : e === 101 ? "\x1B" : e === 32 ? " " : e === 34 ? "\"" : e === 47 ? "/" : e === 92 ? "\\" : e === 78 ? "" : e === 95 ? "\xA0" : e === 76 ? "\u2028" : e === 80 ? "\u2029" : "";
}
function mi(e) {
	return e <= 65535 ? String.fromCharCode(e) : String.fromCharCode((e - 65536 >> 10) + 55296, (e - 65536 & 1023) + 56320);
}
for (var Ae = new Array(256), ve = new Array(256), L = 0; L < 256; L++) Ae[L] = ge(L) ? 1 : 0, ve[L] = ge(L);
function gi(e, n) {
	this.input = e, this.filename = n.filename || null, this.schema = n.schema || pe, this.onWarning = n.onWarning || null, this.legacy = n.legacy || !1, this.json = n.json || !1, this.listener = n.listener || null, this.implicitTypes = this.schema.compiledImplicit, this.typeMap = this.schema.compiledTypeMap, this.length = e.length, this.position = 0, this.line = 0, this.lineStart = 0, this.lineIndent = 0, this.firstTabInLine = -1, this.documents = [];
}
function ye(e, n) {
	var i = {
		name: e.filename,
		buffer: e.input.slice(0, -1),
		position: e.position,
		line: e.line,
		column: e.position - e.lineStart
	};
	return i.snippet = ln(i), new w$1(n, i);
}
function h(e, n) {
	throw ye(e, n);
}
function K$1(e, n) {
	e.onWarning && e.onWarning.call(null, ye(e, n));
}
var Ce = {
	YAML: function(n, i, l) {
		var r, u, o;
		n.version !== null && h(n, "duplication of %YAML directive"), l.length !== 1 && h(n, "YAML directive accepts exactly one argument"), r = /^([0-9]+)\.([0-9]+)$/.exec(l[0]), r === null && h(n, "ill-formed argument of the YAML directive"), u = parseInt(r[1], 10), o = parseInt(r[2], 10), u !== 1 && h(n, "unacceptable YAML version of the document"), n.version = l[0], n.checkLineBreaks = o < 2, o !== 1 && o !== 2 && K$1(n, "unsupported YAML version of the document");
	},
	TAG: function(n, i, l) {
		var r, u;
		l.length !== 2 && h(n, "TAG directive accepts exactly two arguments"), r = l[0], u = l[1], se.test(r) || h(n, "ill-formed tag handle (first argument) of the TAG directive"), T.call(n.tagMap, r) && h(n, "there is a previously declared suffix for \"" + r + "\" tag handle"), xe.test(u) || h(n, "ill-formed tag prefix (second argument) of the TAG directive");
		try {
			u = decodeURIComponent(u);
		} catch {
			h(n, "tag prefix is malformed: " + u);
		}
		n.tagMap[r] = u;
	}
};
function O(e, n, i, l) {
	var r, u, o, f;
	if (n < i) {
		if (f = e.input.slice(n, i), l) for (r = 0, u = f.length; r < u; r += 1) o = f.charCodeAt(r), o === 9 || 32 <= o && o <= 1114111 || h(e, "expected valid JSON character");
		else pi.test(f) && h(e, "the stream contains non-printable characters");
		e.result += f;
	}
}
function _e(e, n, i, l) {
	var r, u, o, f;
	for (y$1.isObject(i) || h(e, "cannot merge mappings; the provided source object is unacceptable"), r = Object.keys(i), o = 0, f = r.length; o < f; o += 1) u = r[o], T.call(n, u) || (n[u] = i[u], l[u] = !0);
}
function N(e, n, i, l, r, u, o, f, c$1) {
	var a, t;
	if (Array.isArray(r)) for (r = Array.prototype.slice.call(r), a = 0, t = r.length; a < t; a += 1) Array.isArray(r[a]) && h(e, "nested arrays are not supported inside keys"), typeof r == "object" && me(r[a]) === "[object Object]" && (r[a] = "[object Object]");
	if (typeof r == "object" && me(r) === "[object Object]" && (r = "[object Object]"), r = String(r), n === null && (n = {}), l === "tag:yaml.org,2002:merge") if (Array.isArray(u)) for (a = 0, t = u.length; a < t; a += 1) _e(e, n, u[a], i);
	else _e(e, n, u, i);
	else !e.json && !T.call(i, r) && T.call(n, r) && (e.line = o || e.line, e.lineStart = f || e.lineStart, e.position = c$1 || e.position, h(e, "duplicated mapping key")), r === "__proto__" ? Object.defineProperty(n, r, {
		configurable: !0,
		enumerable: !0,
		writable: !0,
		value: u
	}) : n[r] = u, delete i[r];
	return n;
}
function z$1(e) {
	var n = e.input.charCodeAt(e.position);
	n === 10 ? e.position++ : n === 13 ? (e.position++, e.input.charCodeAt(e.position) === 10 && e.position++) : h(e, "a line break is expected"), e.line += 1, e.lineStart = e.position, e.firstTabInLine = -1;
}
function v$1(e, n, i) {
	for (var l = 0, r = e.input.charCodeAt(e.position); r !== 0;) {
		for (; I$1(r);) r === 9 && e.firstTabInLine === -1 && (e.firstTabInLine = e.position), r = e.input.charCodeAt(++e.position);
		if (n && r === 35) do
			r = e.input.charCodeAt(++e.position);
		while (r !== 10 && r !== 13 && r !== 0);
		if (S$1(r)) for (z$1(e), r = e.input.charCodeAt(e.position), l++, e.lineIndent = 0; r === 32;) e.lineIndent++, r = e.input.charCodeAt(++e.position);
		else break;
	}
	return i !== -1 && l !== 0 && e.lineIndent < i && K$1(e, "deficient indentation"), l;
}
function q(e) {
	var n = e.position, i;
	return i = e.input.charCodeAt(n), !!((i === 45 || i === 46) && i === e.input.charCodeAt(n + 1) && i === e.input.charCodeAt(n + 2) && (n += 3, i = e.input.charCodeAt(n), i === 0 || F$1(i)));
}
function J(e, n) {
	n === 1 ? e.result += " " : n > 1 && (e.result += y$1.repeat(`
`, n - 1));
}
function Ai(e, n, i) {
	var l, r, u, o, f, c$1, a, t, p$1 = e.kind, d = e.result, s$1;
	if (s$1 = e.input.charCodeAt(e.position), F$1(s$1) || k(s$1) || s$1 === 35 || s$1 === 38 || s$1 === 42 || s$1 === 33 || s$1 === 124 || s$1 === 62 || s$1 === 39 || s$1 === 34 || s$1 === 37 || s$1 === 64 || s$1 === 96 || (s$1 === 63 || s$1 === 45) && (r = e.input.charCodeAt(e.position + 1), F$1(r) || i && k(r))) return !1;
	for (e.kind = "scalar", e.result = "", u = o = e.position, f = !1; s$1 !== 0;) {
		if (s$1 === 58) {
			if (r = e.input.charCodeAt(e.position + 1), F$1(r) || i && k(r)) break;
		} else if (s$1 === 35) {
			if (l = e.input.charCodeAt(e.position - 1), F$1(l)) break;
		} else {
			if (e.position === e.lineStart && q(e) || i && k(s$1)) break;
			if (S$1(s$1)) if (c$1 = e.line, a = e.lineStart, t = e.lineIndent, v$1(e, !1, -1), e.lineIndent >= n) {
				f = !0, s$1 = e.input.charCodeAt(e.position);
				continue;
			} else {
				e.position = o, e.line = c$1, e.lineStart = a, e.lineIndent = t;
				break;
			}
		}
		f && (O(e, u, o, !1), J(e, e.line - c$1), u = o = e.position, f = !1), I$1(s$1) || (o = e.position + 1), s$1 = e.input.charCodeAt(++e.position);
	}
	return O(e, u, o, !1), e.result ? !0 : (e.kind = p$1, e.result = d, !1);
}
function vi(e, n) {
	var i, l, r;
	if (i = e.input.charCodeAt(e.position), i !== 39) return !1;
	for (e.kind = "scalar", e.result = "", e.position++, l = r = e.position; (i = e.input.charCodeAt(e.position)) !== 0;) if (i === 39) if (O(e, l, e.position, !0), i = e.input.charCodeAt(++e.position), i === 39) l = e.position, e.position++, r = e.position;
	else return !0;
	else S$1(i) ? (O(e, l, r, !0), J(e, v$1(e, !1, n)), l = r = e.position) : e.position === e.lineStart && q(e) ? h(e, "unexpected end of the document within a single quoted scalar") : (e.position++, r = e.position);
	h(e, "unexpected end of the stream within a single quoted scalar");
}
function yi(e, n) {
	var i, l, r, u, o, f;
	if (f = e.input.charCodeAt(e.position), f !== 34) return !1;
	for (e.kind = "scalar", e.result = "", e.position++, i = l = e.position; (f = e.input.charCodeAt(e.position)) !== 0;) {
		if (f === 34) return O(e, i, e.position, !0), e.position++, !0;
		if (f === 92) {
			if (O(e, i, e.position, !0), f = e.input.charCodeAt(++e.position), S$1(f)) v$1(e, !1, n);
			else if (f < 256 && Ae[f]) e.result += ve[f], e.position++;
			else if ((o = si(f)) > 0) {
				for (r = o, u = 0; r > 0; r--) f = e.input.charCodeAt(++e.position), (o = di(f)) >= 0 ? u = (u << 4) + o : h(e, "expected hexadecimal character");
				e.result += mi(u), e.position++;
			} else h(e, "unknown escape sequence");
			i = l = e.position;
		} else S$1(f) ? (O(e, i, l, !0), J(e, v$1(e, !1, n)), i = l = e.position) : e.position === e.lineStart && q(e) ? h(e, "unexpected end of the document within a double quoted scalar") : (e.position++, l = e.position);
	}
	h(e, "unexpected end of the stream within a double quoted scalar");
}
function Ci(e, n) {
	var i = !0, l, r, u, o = e.tag, f, c$1 = e.anchor, a, t, p$1, d, s$1, x$1 = Object.create(null), g$1, A$1, b$1, m;
	if (m = e.input.charCodeAt(e.position), m === 91) t = 93, s$1 = !1, f = [];
	else if (m === 123) t = 125, s$1 = !0, f = {};
	else return !1;
	for (e.anchor !== null && (e.anchorMap[e.anchor] = f), m = e.input.charCodeAt(++e.position); m !== 0;) {
		if (v$1(e, !0, n), m = e.input.charCodeAt(e.position), m === t) return e.position++, e.tag = o, e.anchor = c$1, e.kind = s$1 ? "mapping" : "sequence", e.result = f, !0;
		i ? m === 44 && h(e, "expected the node content, but found ','") : h(e, "missed comma between flow collection entries"), A$1 = g$1 = b$1 = null, p$1 = d = !1, m === 63 && (a = e.input.charCodeAt(e.position + 1), F$1(a) && (p$1 = d = !0, e.position++, v$1(e, !0, n))), l = e.line, r = e.lineStart, u = e.position, R$1(e, n, H, !1, !0), A$1 = e.tag, g$1 = e.result, v$1(e, !0, n), m = e.input.charCodeAt(e.position), (d || e.line === l) && m === 58 && (p$1 = !0, m = e.input.charCodeAt(++e.position), v$1(e, !0, n), R$1(e, n, H, !1, !0), b$1 = e.result), s$1 ? N(e, f, x$1, A$1, g$1, b$1, l, r, u) : p$1 ? f.push(N(e, null, x$1, A$1, g$1, b$1, l, r, u)) : f.push(g$1), v$1(e, !0, n), m = e.input.charCodeAt(e.position), m === 44 ? (i = !0, m = e.input.charCodeAt(++e.position)) : i = !1;
	}
	h(e, "unexpected end of the stream within a flow collection");
}
function _i(e, n) {
	var i, l, r = Z$1, u = !1, o = !1, f = n, c$1 = 0, a = !1, t, p$1;
	if (p$1 = e.input.charCodeAt(e.position), p$1 === 124) l = !1;
	else if (p$1 === 62) l = !0;
	else return !1;
	for (e.kind = "scalar", e.result = ""; p$1 !== 0;) if (p$1 = e.input.charCodeAt(++e.position), p$1 === 43 || p$1 === 45) Z$1 === r ? r = p$1 === 43 ? de : ai : h(e, "repeat of a chomping mode identifier");
	else if ((t = xi(p$1)) >= 0) t === 0 ? h(e, "bad explicit indentation width of a block scalar; it cannot be less than one") : o ? h(e, "repeat of an indentation width identifier") : (f = n + t - 1, o = !0);
	else break;
	if (I$1(p$1)) {
		do
			p$1 = e.input.charCodeAt(++e.position);
		while (I$1(p$1));
		if (p$1 === 35) do
			p$1 = e.input.charCodeAt(++e.position);
		while (!S$1(p$1) && p$1 !== 0);
	}
	for (; p$1 !== 0;) {
		for (z$1(e), e.lineIndent = 0, p$1 = e.input.charCodeAt(e.position); (!o || e.lineIndent < f) && p$1 === 32;) e.lineIndent++, p$1 = e.input.charCodeAt(++e.position);
		if (!o && e.lineIndent > f && (f = e.lineIndent), S$1(p$1)) {
			c$1++;
			continue;
		}
		if (e.lineIndent < f) {
			r === de ? e.result += y$1.repeat(`
`, u ? 1 + c$1 : c$1) : r === Z$1 && u && (e.result += `
`);
			break;
		}
		for (l ? I$1(p$1) ? (a = !0, e.result += y$1.repeat(`
`, u ? 1 + c$1 : c$1)) : a ? (a = !1, e.result += y$1.repeat(`
`, c$1 + 1)) : c$1 === 0 ? u && (e.result += " ") : e.result += y$1.repeat(`
`, c$1) : e.result += y$1.repeat(`
`, u ? 1 + c$1 : c$1), u = !0, o = !0, c$1 = 0, i = e.position; !S$1(p$1) && p$1 !== 0;) p$1 = e.input.charCodeAt(++e.position);
		O(e, i, e.position, !1);
	}
	return !0;
}
function we(e, n) {
	var i, l = e.tag, r = e.anchor, u = [], o, f = !1, c$1;
	if (e.firstTabInLine !== -1) return !1;
	for (e.anchor !== null && (e.anchorMap[e.anchor] = u), c$1 = e.input.charCodeAt(e.position); c$1 !== 0 && (e.firstTabInLine !== -1 && (e.position = e.firstTabInLine, h(e, "tab characters must not be used in indentation")), !(c$1 !== 45 || (o = e.input.charCodeAt(e.position + 1), !F$1(o))));) {
		if (f = !0, e.position++, v$1(e, !0, -1) && e.lineIndent <= n) {
			u.push(null), c$1 = e.input.charCodeAt(e.position);
			continue;
		}
		if (i = e.line, R$1(e, n, he, !1, !0), u.push(e.result), v$1(e, !0, -1), c$1 = e.input.charCodeAt(e.position), (e.line === i || e.lineIndent > n) && c$1 !== 0) h(e, "bad indentation of a sequence entry");
		else if (e.lineIndent < n) break;
	}
	return f ? (e.tag = l, e.anchor = r, e.kind = "sequence", e.result = u, !0) : !1;
}
function wi(e, n, i) {
	var l, r, u, o, f, c$1, a = e.tag, t = e.anchor, p$1 = {}, d = Object.create(null), s$1 = null, x$1 = null, g$1 = null, A$1 = !1, b$1 = !1, m;
	if (e.firstTabInLine !== -1) return !1;
	for (e.anchor !== null && (e.anchorMap[e.anchor] = p$1), m = e.input.charCodeAt(e.position); m !== 0;) {
		if (!A$1 && e.firstTabInLine !== -1 && (e.position = e.firstTabInLine, h(e, "tab characters must not be used in indentation")), l = e.input.charCodeAt(e.position + 1), u = e.line, (m === 63 || m === 58) && F$1(l)) m === 63 ? (A$1 && (N(e, p$1, d, s$1, x$1, null, o, f, c$1), s$1 = x$1 = g$1 = null), b$1 = !0, A$1 = !0, r = !0) : A$1 ? (A$1 = !1, r = !0) : h(e, "incomplete explicit mapping pair; a key node is missed; or followed by a non-tabulated empty line"), e.position += 1, m = l;
		else {
			if (o = e.line, f = e.lineStart, c$1 = e.position, !R$1(e, i, te, !1, !0)) break;
			if (e.line === u) {
				for (m = e.input.charCodeAt(e.position); I$1(m);) m = e.input.charCodeAt(++e.position);
				if (m === 58) m = e.input.charCodeAt(++e.position), F$1(m) || h(e, "a whitespace character is expected after the key-value separator within a block mapping"), A$1 && (N(e, p$1, d, s$1, x$1, null, o, f, c$1), s$1 = x$1 = g$1 = null), b$1 = !0, A$1 = !1, r = !1, s$1 = e.tag, x$1 = e.result;
				else if (b$1) h(e, "can not read an implicit mapping pair; a colon is missed");
				else return e.tag = a, e.anchor = t, !0;
			} else if (b$1) h(e, "can not read a block mapping entry; a multiline key may not be an implicit key");
			else return e.tag = a, e.anchor = t, !0;
		}
		if ((e.line === u || e.lineIndent > n) && (A$1 && (o = e.line, f = e.lineStart, c$1 = e.position), R$1(e, n, U$1, !0, r) && (A$1 ? x$1 = e.result : g$1 = e.result), A$1 || (N(e, p$1, d, s$1, x$1, g$1, o, f, c$1), s$1 = x$1 = g$1 = null), v$1(e, !0, -1), m = e.input.charCodeAt(e.position)), (e.line === u || e.lineIndent > n) && m !== 0) h(e, "bad indentation of a mapping entry");
		else if (e.lineIndent < n) break;
	}
	return A$1 && N(e, p$1, d, s$1, x$1, null, o, f, c$1), b$1 && (e.tag = a, e.anchor = t, e.kind = "mapping", e.result = p$1), b$1;
}
function Fi(e) {
	var n, i = !1, l = !1, r, u, o;
	if (o = e.input.charCodeAt(e.position), o !== 33) return !1;
	if (e.tag !== null && h(e, "duplication of a tag property"), o = e.input.charCodeAt(++e.position), o === 60 ? (i = !0, o = e.input.charCodeAt(++e.position)) : o === 33 ? (l = !0, r = "!!", o = e.input.charCodeAt(++e.position)) : r = "!", n = e.position, i) {
		do
			o = e.input.charCodeAt(++e.position);
		while (o !== 0 && o !== 62);
		e.position < e.length ? (u = e.input.slice(n, e.position), o = e.input.charCodeAt(++e.position)) : h(e, "unexpected end of the stream within a verbatim tag");
	} else {
		for (; o !== 0 && !F$1(o);) o === 33 && (l ? h(e, "tag suffix cannot contain exclamation marks") : (r = e.input.slice(n - 1, e.position + 1), se.test(r) || h(e, "named tag handle cannot contain such characters"), l = !0, n = e.position + 1)), o = e.input.charCodeAt(++e.position);
		u = e.input.slice(n, e.position), hi.test(u) && h(e, "tag suffix cannot contain flow indicator characters");
	}
	u && !xe.test(u) && h(e, "tag name cannot contain such characters: " + u);
	try {
		u = decodeURIComponent(u);
	} catch {
		h(e, "tag name is malformed: " + u);
	}
	return i ? e.tag = u : T.call(e.tagMap, r) ? e.tag = e.tagMap[r] + u : r === "!" ? e.tag = "!" + u : r === "!!" ? e.tag = "tag:yaml.org,2002:" + u : h(e, "undeclared tag handle \"" + r + "\""), !0;
}
function bi(e) {
	var n, i;
	if (i = e.input.charCodeAt(e.position), i !== 38) return !1;
	for (e.anchor !== null && h(e, "duplication of an anchor property"), i = e.input.charCodeAt(++e.position), n = e.position; i !== 0 && !F$1(i) && !k(i);) i = e.input.charCodeAt(++e.position);
	return e.position === n && h(e, "name of an anchor node must contain at least one character"), e.anchor = e.input.slice(n, e.position), !0;
}
function Si(e) {
	var n, i, l;
	if (l = e.input.charCodeAt(e.position), l !== 42) return !1;
	for (l = e.input.charCodeAt(++e.position), n = e.position; l !== 0 && !F$1(l) && !k(l);) l = e.input.charCodeAt(++e.position);
	return e.position === n && h(e, "name of an alias node must contain at least one character"), i = e.input.slice(n, e.position), T.call(e.anchorMap, i) || h(e, "unidentified alias \"" + i + "\""), e.result = e.anchorMap[i], v$1(e, !0, -1), !0;
}
function R$1(e, n, i, l, r) {
	var u, o, f, c$1 = 1, a = !1, t = !1, p$1, d, s$1, x$1, g$1, A$1;
	if (e.listener !== null && e.listener("open", e), e.tag = null, e.anchor = null, e.kind = null, e.result = null, u = o = f = U$1 === i || he === i, l && v$1(e, !0, -1) && (a = !0, e.lineIndent > n ? c$1 = 1 : e.lineIndent === n ? c$1 = 0 : e.lineIndent < n && (c$1 = -1)), c$1 === 1) for (; Fi(e) || bi(e);) v$1(e, !0, -1) ? (a = !0, f = u, e.lineIndent > n ? c$1 = 1 : e.lineIndent === n ? c$1 = 0 : e.lineIndent < n && (c$1 = -1)) : f = !1;
	if (f && (f = a || r), (c$1 === 1 || U$1 === i) && (H === i || te === i ? g$1 = n : g$1 = n + 1, A$1 = e.position - e.lineStart, c$1 === 1 ? f && (we(e, A$1) || wi(e, A$1, g$1)) || Ci(e, g$1) ? t = !0 : (o && _i(e, g$1) || vi(e, g$1) || yi(e, g$1) ? t = !0 : Si(e) ? (t = !0, (e.tag !== null || e.anchor !== null) && h(e, "alias node should not have any properties")) : Ai(e, g$1, H === i) && (t = !0, e.tag === null && (e.tag = "?")), e.anchor !== null && (e.anchorMap[e.anchor] = e.result)) : c$1 === 0 && (t = f && we(e, A$1))), e.tag === null) e.anchor !== null && (e.anchorMap[e.anchor] = e.result);
	else if (e.tag === "?") {
		for (e.result !== null && e.kind !== "scalar" && h(e, "unacceptable node kind for !<?> tag; it should be \"scalar\", not \"" + e.kind + "\""), p$1 = 0, d = e.implicitTypes.length; p$1 < d; p$1 += 1) if (x$1 = e.implicitTypes[p$1], x$1.resolve(e.result)) {
			e.result = x$1.construct(e.result), e.tag = x$1.tag, e.anchor !== null && (e.anchorMap[e.anchor] = e.result);
			break;
		}
	} else if (e.tag !== "!") {
		if (T.call(e.typeMap[e.kind || "fallback"], e.tag)) x$1 = e.typeMap[e.kind || "fallback"][e.tag];
		else for (x$1 = null, s$1 = e.typeMap.multi[e.kind || "fallback"], p$1 = 0, d = s$1.length; p$1 < d; p$1 += 1) if (e.tag.slice(0, s$1[p$1].tag.length) === s$1[p$1].tag) {
			x$1 = s$1[p$1];
			break;
		}
		x$1 || h(e, "unknown tag !<" + e.tag + ">"), e.result !== null && x$1.kind !== e.kind && h(e, "unacceptable node kind for !<" + e.tag + "> tag; it should be \"" + x$1.kind + "\", not \"" + e.kind + "\""), x$1.resolve(e.result, e.tag) ? (e.result = x$1.construct(e.result, e.tag), e.anchor !== null && (e.anchorMap[e.anchor] = e.result)) : h(e, "cannot resolve a node with !<" + e.tag + "> explicit tag");
	}
	return e.listener !== null && e.listener("close", e), e.tag !== null || e.anchor !== null || t;
}
function Ei(e) {
	var n = e.position, i, l, r, u = !1, o;
	for (e.version = null, e.checkLineBreaks = e.legacy, e.tagMap = Object.create(null), e.anchorMap = Object.create(null); (o = e.input.charCodeAt(e.position)) !== 0 && (v$1(e, !0, -1), o = e.input.charCodeAt(e.position), !(e.lineIndent > 0 || o !== 37));) {
		for (u = !0, o = e.input.charCodeAt(++e.position), i = e.position; o !== 0 && !F$1(o);) o = e.input.charCodeAt(++e.position);
		for (l = e.input.slice(i, e.position), r = [], l.length < 1 && h(e, "directive name must not be less than one character in length"); o !== 0;) {
			for (; I$1(o);) o = e.input.charCodeAt(++e.position);
			if (o === 35) {
				do
					o = e.input.charCodeAt(++e.position);
				while (o !== 0 && !S$1(o));
				break;
			}
			if (S$1(o)) break;
			for (i = e.position; o !== 0 && !F$1(o);) o = e.input.charCodeAt(++e.position);
			r.push(e.input.slice(i, e.position));
		}
		o !== 0 && z$1(e), T.call(Ce, l) ? Ce[l](e, l, r) : K$1(e, "unknown document directive \"" + l + "\"");
	}
	if (v$1(e, !0, -1), e.lineIndent === 0 && e.input.charCodeAt(e.position) === 45 && e.input.charCodeAt(e.position + 1) === 45 && e.input.charCodeAt(e.position + 2) === 45 ? (e.position += 3, v$1(e, !0, -1)) : u && h(e, "directives end mark is expected"), R$1(e, e.lineIndent - 1, U$1, !1, !0), v$1(e, !0, -1), e.checkLineBreaks && ti.test(e.input.slice(n, e.position)) && K$1(e, "non-ASCII line breaks are interpreted as content"), e.documents.push(e.result), e.position === e.lineStart && q(e)) {
		e.input.charCodeAt(e.position) === 46 && (e.position += 3, v$1(e, !0, -1));
		return;
	}
	if (e.position < e.length - 1) h(e, "end of the stream or a document separator is expected");
	else return;
}
function Ti(e, n) {
	e = String(e), n = n || {}, e.length !== 0 && (e.charCodeAt(e.length - 1) !== 10 && e.charCodeAt(e.length - 1) !== 13 && (e += `
`), e.charCodeAt(0) === 65279 && (e = e.slice(1)));
	var i = new gi(e, n), l = e.indexOf("\0");
	for (l !== -1 && (i.position = l, h(i, "null byte is not allowed in input")), i.input += "\0"; i.input.charCodeAt(i.position) === 32;) i.lineIndent += 1, i.position += 1;
	for (; i.position < i.length - 1;) Ei(i);
	return i.documents;
}
function Oi(e, n) {
	var i = Ti(e, n);
	if (i.length !== 0) {
		if (i.length === 1) return i[0];
		throw new w$1("expected a single document in the stream, but found more");
	}
}
var ki = { load: Oi }, Fe = Object.prototype.toString, be = Object.prototype.hasOwnProperty, ee = 65279, Li = 9, Y = 10, Ni = 13, Ri = 32, Di = 33, Mi = 34, ne = 35, Yi = 37, Bi = 38, Pi = 39, ji = 42, Se = 44, Hi = 45, G$1 = 58, Ui = 61, Ki = 62, qi = 63, Gi = 64, Ee = 91, Te = 93, Wi = 96, Oe = 123, $i = 124, Ie = 125, _ = {};
_[0] = "\\0", _[7] = "\\a", _[8] = "\\b", _[9] = "\\t", _[10] = "\\n", _[11] = "\\v", _[12] = "\\f", _[13] = "\\r", _[27] = "\\e", _[34] = "\\\"", _[92] = "\\\\", _[133] = "\\N", _[160] = "\\_", _[8232] = "\\L", _[8233] = "\\P";
var Qi = [
	"y",
	"Y",
	"yes",
	"Yes",
	"YES",
	"on",
	"On",
	"ON",
	"n",
	"N",
	"no",
	"No",
	"NO",
	"off",
	"Off",
	"OFF"
], Vi = /^[-+]?[0-9_]+(?::[0-9_]+)+(?:\.[0-9_]*)?$/;
function Xi(e, n) {
	var i, l, r, u, o, f, c$1;
	if (n === null) return {};
	for (i = {}, l = Object.keys(n), r = 0, u = l.length; r < u; r += 1) o = l[r], f = String(n[o]), o.slice(0, 2) === "!!" && (o = "tag:yaml.org,2002:" + o.slice(2)), c$1 = e.compiledTypeMap.fallback[o], c$1 && be.call(c$1.styleAliases, f) && (f = c$1.styleAliases[f]), i[o] = f;
	return i;
}
function Zi(e) {
	var n, i, l;
	if (n = e.toString(16).toUpperCase(), e <= 255) i = "x", l = 2;
	else if (e <= 65535) i = "u", l = 4;
	else if (e <= 4294967295) i = "U", l = 8;
	else throw new w$1("code point within a string may not be greater than 0xFFFFFFFF");
	return "\\" + i + y$1.repeat("0", l - n.length) + n;
}
var zi = 1, B = 2;
function Ji(e) {
	this.schema = e.schema || pe, this.indent = Math.max(1, e.indent || 2), this.noArrayIndent = e.noArrayIndent || !1, this.skipInvalid = e.skipInvalid || !1, this.flowLevel = y$1.isNothing(e.flowLevel) ? -1 : e.flowLevel, this.styleMap = Xi(this.schema, e.styles || null), this.sortKeys = e.sortKeys || !1, this.lineWidth = e.lineWidth || 80, this.noRefs = e.noRefs || !1, this.noCompatMode = e.noCompatMode || !1, this.condenseFlow = e.condenseFlow || !1, this.quotingType = e.quotingType === "\"" ? B : zi, this.forceQuotes = e.forceQuotes || !1, this.replacer = typeof e.replacer == "function" ? e.replacer : null, this.implicitTypes = this.schema.compiledImplicit, this.explicitTypes = this.schema.compiledExplicit, this.tag = null, this.result = "", this.duplicates = [], this.usedDuplicates = null;
}
function ke(e, n) {
	for (var i = y$1.repeat(" ", n), l = 0, r = -1, u = "", o, f = e.length; l < f;) r = e.indexOf(`
`, l), r === -1 ? (o = e.slice(l), l = f) : (o = e.slice(l, r + 1), l = r + 1), o.length && o !== `
` && (u += i), u += o;
	return u;
}
function ie(e, n) {
	return `
` + y$1.repeat(" ", e.indent * n);
}
function er(e, n) {
	var i, l, r;
	for (i = 0, l = e.implicitTypes.length; i < l; i += 1) if (r = e.implicitTypes[i], r.resolve(n)) return !0;
	return !1;
}
function W(e) {
	return e === Ri || e === Li;
}
function P(e) {
	return 32 <= e && e <= 126 || 161 <= e && e <= 55295 && e !== 8232 && e !== 8233 || 57344 <= e && e <= 65533 && e !== ee || 65536 <= e && e <= 1114111;
}
function Le(e) {
	return P(e) && e !== ee && e !== Ni && e !== Y;
}
function Ne(e, n, i) {
	var l = Le(e), r = l && !W(e);
	return (i ? l : l && e !== Se && e !== Ee && e !== Te && e !== Oe && e !== Ie) && e !== ne && !(n === G$1 && !r) || Le(n) && !W(n) && e === ne || n === G$1 && r;
}
function nr(e) {
	return P(e) && e !== ee && !W(e) && e !== Hi && e !== qi && e !== G$1 && e !== Se && e !== Ee && e !== Te && e !== Oe && e !== Ie && e !== ne && e !== Bi && e !== ji && e !== Di && e !== $i && e !== Ui && e !== Ki && e !== Pi && e !== Mi && e !== Yi && e !== Gi && e !== Wi;
}
function ir(e) {
	return !W(e) && e !== G$1;
}
function j$1(e, n) {
	var i = e.charCodeAt(n), l;
	return i >= 55296 && i <= 56319 && n + 1 < e.length && (l = e.charCodeAt(n + 1), l >= 56320 && l <= 57343) ? (i - 55296) * 1024 + l - 56320 + 65536 : i;
}
function Re(e) {
	return /^\n* /.test(e);
}
var De = 1, re = 2, Me = 3, Ye = 4, D = 5;
function rr(e, n, i, l, r, u, o, f) {
	var c$1, a = 0, t = null, p$1 = !1, d = !1, s$1 = l !== -1, x$1 = -1, g$1 = nr(j$1(e, 0)) && ir(j$1(e, e.length - 1));
	if (n || o) for (c$1 = 0; c$1 < e.length; a >= 65536 ? c$1 += 2 : c$1++) {
		if (a = j$1(e, c$1), !P(a)) return D;
		g$1 = g$1 && Ne(a, t, f), t = a;
	}
	else {
		for (c$1 = 0; c$1 < e.length; a >= 65536 ? c$1 += 2 : c$1++) {
			if (a = j$1(e, c$1), a === Y) p$1 = !0, s$1 && (d = d || c$1 - x$1 - 1 > l && e[x$1 + 1] !== " ", x$1 = c$1);
			else if (!P(a)) return D;
			g$1 = g$1 && Ne(a, t, f), t = a;
		}
		d = d || s$1 && c$1 - x$1 - 1 > l && e[x$1 + 1] !== " ";
	}
	return !p$1 && !d ? g$1 && !o && !r(e) ? De : u === B ? D : re : i > 9 && Re(e) ? D : o ? u === B ? D : re : d ? Ye : Me;
}
function lr(e, n, i, l, r) {
	e.dump = function() {
		if (n.length === 0) return e.quotingType === B ? "\"\"" : "''";
		if (!e.noCompatMode && (Qi.indexOf(n) !== -1 || Vi.test(n))) return e.quotingType === B ? "\"" + n + "\"" : "'" + n + "'";
		var u = e.indent * Math.max(1, i), o = e.lineWidth === -1 ? -1 : Math.max(Math.min(e.lineWidth, 40), e.lineWidth - u), f = l || e.flowLevel > -1 && i >= e.flowLevel;
		function c$1(a) {
			return er(e, a);
		}
		switch (rr(n, f, e.indent, o, c$1, e.quotingType, e.forceQuotes && !l, r)) {
			case De: return n;
			case re: return "'" + n.replace(/'/g, "''") + "'";
			case Me: return "|" + Be(n, e.indent) + Pe(ke(n, u));
			case Ye: return ">" + Be(n, e.indent) + Pe(ke(or(n, o), u));
			case D: return "\"" + ur(n) + "\"";
			default: throw new w$1("impossible error: invalid scalar style");
		}
	}();
}
function Be(e, n) {
	var i = Re(e) ? String(n) : "", l = e[e.length - 1] === `
`;
	return i + (l && (e[e.length - 2] === `
` || e === `
`) ? "+" : l ? "" : "-") + `
`;
}
function Pe(e) {
	return e[e.length - 1] === `
` ? e.slice(0, -1) : e;
}
function or(e, n) {
	for (var i = /(\n+)([^\n]*)/g, l = function() {
		var a = e.indexOf(`
`);
		return a = a !== -1 ? a : e.length, i.lastIndex = a, je(e.slice(0, a), n);
	}(), r = e[0] === `
` || e[0] === " ", u, o; o = i.exec(e);) {
		var f = o[1], c$1 = o[2];
		u = c$1[0] === " ", l += f + (!r && !u && c$1 !== "" ? `
` : "") + je(c$1, n), r = u;
	}
	return l;
}
function je(e, n) {
	if (e === "" || e[0] === " ") return e;
	for (var i = / [^ ]/g, l, r = 0, u, o = 0, f = 0, c$1 = ""; l = i.exec(e);) f = l.index, f - r > n && (u = o > r ? o : f, c$1 += `
` + e.slice(r, u), r = u + 1), o = f;
	return c$1 += `
`, e.length - r > n && o > r ? c$1 += e.slice(r, o) + `
` + e.slice(o + 1) : c$1 += e.slice(r), c$1.slice(1);
}
function ur(e) {
	for (var n = "", i = 0, l, r = 0; r < e.length; i >= 65536 ? r += 2 : r++) i = j$1(e, r), l = _[i], !l && P(i) ? (n += e[r], i >= 65536 && (n += e[r + 1])) : n += l || Zi(i);
	return n;
}
function fr(e, n, i) {
	var l = "", r = e.tag, u, o, f;
	for (u = 0, o = i.length; u < o; u += 1) f = i[u], e.replacer && (f = e.replacer.call(i, String(u), f)), (E(e, n, f, !1, !1) || typeof f > "u" && E(e, n, null, !1, !1)) && (l !== "" && (l += "," + (e.condenseFlow ? "" : " ")), l += e.dump);
	e.tag = r, e.dump = "[" + l + "]";
}
function He(e, n, i, l) {
	var r = "", u = e.tag, o, f, c$1;
	for (o = 0, f = i.length; o < f; o += 1) c$1 = i[o], e.replacer && (c$1 = e.replacer.call(i, String(o), c$1)), (E(e, n + 1, c$1, !0, !0, !1, !0) || typeof c$1 > "u" && E(e, n + 1, null, !0, !0, !1, !0)) && ((!l || r !== "") && (r += ie(e, n)), e.dump && Y === e.dump.charCodeAt(0) ? r += "-" : r += "- ", r += e.dump);
	e.tag = u, e.dump = r || "[]";
}
function cr(e, n, i) {
	var l = "", r = e.tag, u = Object.keys(i), o, f, c$1, a, t;
	for (o = 0, f = u.length; o < f; o += 1) t = "", l !== "" && (t += ", "), e.condenseFlow && (t += "\""), c$1 = u[o], a = i[c$1], e.replacer && (a = e.replacer.call(i, c$1, a)), E(e, n, c$1, !1, !1) && (e.dump.length > 1024 && (t += "? "), t += e.dump + (e.condenseFlow ? "\"" : "") + ":" + (e.condenseFlow ? "" : " "), E(e, n, a, !1, !1) && (t += e.dump, l += t));
	e.tag = r, e.dump = "{" + l + "}";
}
function ar(e, n, i, l) {
	var r = "", u = e.tag, o = Object.keys(i), f, c$1, a, t, p$1, d;
	if (e.sortKeys === !0) o.sort();
	else if (typeof e.sortKeys == "function") o.sort(e.sortKeys);
	else if (e.sortKeys) throw new w$1("sortKeys must be a boolean or a function");
	for (f = 0, c$1 = o.length; f < c$1; f += 1) d = "", (!l || r !== "") && (d += ie(e, n)), a = o[f], t = i[a], e.replacer && (t = e.replacer.call(i, a, t)), E(e, n + 1, a, !0, !0, !0) && (p$1 = e.tag !== null && e.tag !== "?" || e.dump && e.dump.length > 1024, p$1 && (e.dump && Y === e.dump.charCodeAt(0) ? d += "?" : d += "? "), d += e.dump, p$1 && (d += ie(e, n)), E(e, n + 1, t, !0, p$1) && (e.dump && Y === e.dump.charCodeAt(0) ? d += ":" : d += ": ", d += e.dump, r += d));
	e.tag = u, e.dump = r || "{}";
}
function Ue(e, n, i) {
	var l, r, u, o, f, c$1;
	for (r = i ? e.explicitTypes : e.implicitTypes, u = 0, o = r.length; u < o; u += 1) if (f = r[u], (f.instanceOf || f.predicate) && (!f.instanceOf || typeof n == "object" && n instanceof f.instanceOf) && (!f.predicate || f.predicate(n))) {
		if (i ? f.multi && f.representName ? e.tag = f.representName(n) : e.tag = f.tag : e.tag = "?", f.represent) {
			if (c$1 = e.styleMap[f.tag] || f.defaultStyle, Fe.call(f.represent) === "[object Function]") l = f.represent(n, c$1);
			else if (be.call(f.represent, c$1)) l = f.represent[c$1](n, c$1);
			else throw new w$1("!<" + f.tag + "> tag resolver accepts not \"" + c$1 + "\" style");
			e.dump = l;
		}
		return !0;
	}
	return !1;
}
function E(e, n, i, l, r, u, o) {
	e.tag = null, e.dump = i, Ue(e, i, !1) || Ue(e, i, !0);
	var f = Fe.call(e.dump), c$1 = l, a;
	l && (l = e.flowLevel < 0 || e.flowLevel > n);
	var t = f === "[object Object]" || f === "[object Array]", p$1, d;
	if (t && (p$1 = e.duplicates.indexOf(i), d = p$1 !== -1), (e.tag !== null && e.tag !== "?" || d || e.indent !== 2 && n > 0) && (r = !1), d && e.usedDuplicates[p$1]) e.dump = "*ref_" + p$1;
	else {
		if (t && d && !e.usedDuplicates[p$1] && (e.usedDuplicates[p$1] = !0), f === "[object Object]") l && Object.keys(e.dump).length !== 0 ? (ar(e, n, e.dump, r), d && (e.dump = "&ref_" + p$1 + e.dump)) : (cr(e, n, e.dump), d && (e.dump = "&ref_" + p$1 + " " + e.dump));
		else if (f === "[object Array]") l && e.dump.length !== 0 ? (e.noArrayIndent && !o && n > 0 ? He(e, n - 1, e.dump, r) : He(e, n, e.dump, r), d && (e.dump = "&ref_" + p$1 + e.dump)) : (fr(e, n, e.dump), d && (e.dump = "&ref_" + p$1 + " " + e.dump));
		else if (f === "[object String]") e.tag !== "?" && lr(e, e.dump, n, u, c$1);
		else {
			if (f === "[object Undefined]") return !1;
			if (e.skipInvalid) return !1;
			throw new w$1("unacceptable kind of an object to dump " + f);
		}
		e.tag !== null && e.tag !== "?" && (a = encodeURI(e.tag[0] === "!" ? e.tag.slice(1) : e.tag).replace(/!/g, "%21"), e.tag[0] === "!" ? a = "!" + a : a.slice(0, 18) === "tag:yaml.org,2002:" ? a = "!!" + a.slice(18) : a = "!<" + a + ">", e.dump = a + " " + e.dump);
	}
	return !0;
}
function pr(e, n) {
	var i = [], l = [], r, u;
	for (le(e, i, l), r = 0, u = l.length; r < u; r += 1) n.duplicates.push(i[l[r]]);
	n.usedDuplicates = new Array(u);
}
function le(e, n, i) {
	var l, r, u;
	if (e !== null && typeof e == "object") if (r = n.indexOf(e), r !== -1) i.indexOf(r) === -1 && i.push(r);
	else if (n.push(e), Array.isArray(e)) for (r = 0, u = e.length; r < u; r += 1) le(e[r], n, i);
	else for (l = Object.keys(e), r = 0, u = l.length; r < u; r += 1) le(e[l[r]], n, i);
}
function tr(e, n) {
	n = n || {};
	var i = new Ji(n);
	i.noRefs || pr(e, i);
	var l = e;
	return i.replacer && (l = i.replacer.call({ "": l }, "", l)), E(i, 0, l, !0, !0) ? i.dump + `
` : "";
}
var dr = { dump: tr }, sr = ki.load, xr = dr.dump;
function mr(e, n) {
	const i = sr(e, n);
	return N$1(e, i, n), i;
}
function gr(e, n) {
	const i = C$2(e, {}), r = xr(e, {
		indent: typeof i.indent == "string" ? i.indent.length : i.indent,
		...n
	});
	return i.whitespace.start + r.trim() + i.whitespace.end;
}

//#endregion
//#region node_modules/.pnpm/confbox@0.2.2/node_modules/confbox/dist/toml.mjs
var toml_exports = /* @__PURE__ */ __exportAll({ parseTOML: () => Q });
/*!
* Copyright (c) Squirrel Chat et al., All rights reserved.
* SPDX-License-Identifier: BSD-3-Clause
*
* Redistribution and use in source and binary forms, with or without
* modification, are permitted provided that the following conditions are met:
*
* 1. Redistributions of source code must retain the above copyright notice, this
*    list of conditions and the following disclaimer.
* 2. Redistributions in binary form must reproduce the above copyright notice,
*    this list of conditions and the following disclaimer in the
*    documentation and/or other materials provided with the distribution.
* 3. Neither the name of the copyright holder nor the names of its contributors
*    may be used to endorse or promote products derived from this software without
*    specific prior written permission.
*
* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
* ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
* DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
* FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
* DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
* CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
*/ function v(e, n) {
	let t = e.slice(0, n).split(/\r\n|\n|\r/g);
	return [t.length, t.pop().length + 1];
}
function C(e, n, t) {
	let l = e.split(/\r\n|\n|\r/g), r = "", i = (Math.log10(n + 1) | 0) + 1;
	for (let o = n - 1; o <= n + 1; o++) {
		let f = l[o - 1];
		f && (r += o.toString().padEnd(i, " "), r += ":  ", r += f, r += `
`, o === n && (r += " ".repeat(i + t + 2), r += `^
`));
	}
	return r;
}
var c = class extends Error {
	line;
	column;
	codeblock;
	constructor(n, t) {
		const [l, r] = v(t.toml, t.ptr), i = C(t.toml, l, r);
		super(`Invalid TOML document: ${n}

${i}`, t), this.line = l, this.column = r, this.codeblock = i;
	}
};
/*!
* Copyright (c) Squirrel Chat et al., All rights reserved.
* SPDX-License-Identifier: BSD-3-Clause
*
* Redistribution and use in source and binary forms, with or without
* modification, are permitted provided that the following conditions are met:
*
* 1. Redistributions of source code must retain the above copyright notice, this
*    list of conditions and the following disclaimer.
* 2. Redistributions in binary form must reproduce the above copyright notice,
*    this list of conditions and the following disclaimer in the
*    documentation and/or other materials provided with the distribution.
* 3. Neither the name of the copyright holder nor the names of its contributors
*    may be used to endorse or promote products derived from this software without
*    specific prior written permission.
*
* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
* ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
* DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
* FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
* DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
* CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
*/ function g(e, n = 0, t = e.length) {
	let l = e.indexOf(`
`, n);
	return e[l - 1] === "\r" && l--, l <= t ? l : -1;
}
function y(e, n) {
	for (let t = n; t < e.length; t++) {
		let l = e[t];
		if (l === `
`) return t;
		if (l === "\r" && e[t + 1] === `
`) return t + 1;
		if (l < " " && l !== "	" || l === "") throw new c("control characters are not allowed in comments", {
			toml: e,
			ptr: n
		});
	}
	return e.length;
}
function s(e, n, t, l) {
	let r;
	for (; (r = e[n]) === " " || r === "	" || !t && (r === `
` || r === "\r" && e[n + 1] === `
`);) n++;
	return l || r !== "#" ? n : s(e, y(e, n), t);
}
function A(e, n, t, l, r = !1) {
	if (!l) return n = g(e, n), n < 0 ? e.length : n;
	for (let i = n; i < e.length; i++) {
		let o = e[i];
		if (o === "#") i = g(e, i);
		else {
			if (o === t) return i + 1;
			if (o === l) return i;
			if (r && (o === `
` || o === "\r" && e[i + 1] === `
`)) return i;
		}
	}
	throw new c("cannot find end of structure", {
		toml: e,
		ptr: n
	});
}
function S(e, n) {
	let t = e[n], l = t === e[n + 1] && e[n + 1] === e[n + 2] ? e.slice(n, n + 3) : t;
	n += l.length - 1;
	do
		n = e.indexOf(l, ++n);
	while (n > -1 && t !== "'" && e[n - 1] === "\\" && e[n - 2] !== "\\");
	return n > -1 && (n += l.length, l.length > 1 && (e[n] === t && n++, e[n] === t && n++)), n;
}
/*!
* Copyright (c) Squirrel Chat et al., All rights reserved.
* SPDX-License-Identifier: BSD-3-Clause
*
* Redistribution and use in source and binary forms, with or without
* modification, are permitted provided that the following conditions are met:
*
* 1. Redistributions of source code must retain the above copyright notice, this
*    list of conditions and the following disclaimer.
* 2. Redistributions in binary form must reproduce the above copyright notice,
*    this list of conditions and the following disclaimer in the
*    documentation and/or other materials provided with the distribution.
* 3. Neither the name of the copyright holder nor the names of its contributors
*    may be used to endorse or promote products derived from this software without
*    specific prior written permission.
*
* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
* ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
* DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
* FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
* DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
* CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
*/ let R = /^(\d{4}-\d{2}-\d{2})?[T ]?(?:(\d{2}):\d{2}:\d{2}(?:\.\d+)?)?(Z|[-+]\d{2}:\d{2})?$/i;
var w = class w extends Date {
	#n = !1;
	#t = !1;
	#e = null;
	constructor(n) {
		let t = !0, l = !0, r = "Z";
		if (typeof n == "string") {
			let i = n.match(R);
			i ? (i[1] || (t = !1, n = `0000-01-01T${n}`), l = !!i[2], i[2] && +i[2] > 23 ? n = "" : (r = i[3] || null, n = n.toUpperCase(), !r && l && (n += "Z"))) : n = "";
		}
		super(n), isNaN(this.getTime()) || (this.#n = t, this.#t = l, this.#e = r);
	}
	isDateTime() {
		return this.#n && this.#t;
	}
	isLocal() {
		return !this.#n || !this.#t || !this.#e;
	}
	isDate() {
		return this.#n && !this.#t;
	}
	isTime() {
		return this.#t && !this.#n;
	}
	isValid() {
		return this.#n || this.#t;
	}
	toISOString() {
		let n = super.toISOString();
		if (this.isDate()) return n.slice(0, 10);
		if (this.isTime()) return n.slice(11, 23);
		if (this.#e === null) return n.slice(0, -1);
		if (this.#e === "Z") return n;
		let t = +this.#e.slice(1, 3) * 60 + +this.#e.slice(4, 6);
		return t = this.#e[0] === "-" ? t : -t, (/* @__PURE__ */ new Date(this.getTime() - t * 6e4)).toISOString().slice(0, -1) + this.#e;
	}
	static wrapAsOffsetDateTime(n, t = "Z") {
		let l = new w(n);
		return l.#e = t, l;
	}
	static wrapAsLocalDateTime(n) {
		let t = new w(n);
		return t.#e = null, t;
	}
	static wrapAsLocalDate(n) {
		let t = new w(n);
		return t.#t = !1, t.#e = null, t;
	}
	static wrapAsLocalTime(n) {
		let t = new w(n);
		return t.#n = !1, t.#e = null, t;
	}
};
/*!
* Copyright (c) Squirrel Chat et al., All rights reserved.
* SPDX-License-Identifier: BSD-3-Clause
*
* Redistribution and use in source and binary forms, with or without
* modification, are permitted provided that the following conditions are met:
*
* 1. Redistributions of source code must retain the above copyright notice, this
*    list of conditions and the following disclaimer.
* 2. Redistributions in binary form must reproduce the above copyright notice,
*    this list of conditions and the following disclaimer in the
*    documentation and/or other materials provided with the distribution.
* 3. Neither the name of the copyright holder nor the names of its contributors
*    may be used to endorse or promote products derived from this software without
*    specific prior written permission.
*
* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
* ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
* DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
* FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
* DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
* CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
*/ let M = /^((0x[0-9a-fA-F](_?[0-9a-fA-F])*)|(([+-]|0[ob])?\d(_?\d)*))$/, Z = /^[+-]?\d(_?\d)*(\.\d(_?\d)*)?([eE][+-]?\d(_?\d)*)?$/, j = /^[+-]?0[0-9_]/, z = /^[0-9a-f]{4,8}$/i, I = {
	b: "\b",
	t: "	",
	n: `
`,
	f: "\f",
	r: "\r",
	"\"": "\"",
	"\\": "\\"
};
function $(e, n = 0, t = e.length) {
	let l = e[n] === "'", r = e[n++] === e[n] && e[n] === e[n + 1];
	r && (t -= 2, e[n += 2] === "\r" && n++, e[n] === `
` && n++);
	let i = 0, o, f = "", a = n;
	for (; n < t - 1;) {
		let u = e[n++];
		if (u === `
` || u === "\r" && e[n] === `
`) {
			if (!r) throw new c("newlines are not allowed in strings", {
				toml: e,
				ptr: n - 1
			});
		} else if (u < " " && u !== "	" || u === "") throw new c("control characters are not allowed in strings", {
			toml: e,
			ptr: n - 1
		});
		if (o) {
			if (o = !1, u === "u" || u === "U") {
				let d = e.slice(n, n += u === "u" ? 4 : 8);
				if (!z.test(d)) throw new c("invalid unicode escape", {
					toml: e,
					ptr: i
				});
				try {
					f += String.fromCodePoint(parseInt(d, 16));
				} catch {
					throw new c("invalid unicode escape", {
						toml: e,
						ptr: i
					});
				}
			} else if (r && (u === `
` || u === " " || u === "	" || u === "\r")) {
				if (n = s(e, n - 1, !0), e[n] !== `
` && e[n] !== "\r") throw new c("invalid escape: only line-ending whitespace may be escaped", {
					toml: e,
					ptr: i
				});
				n = s(e, n);
			} else if (u in I) f += I[u];
			else throw new c("unrecognized escape sequence", {
				toml: e,
				ptr: i
			});
			a = n;
		} else !l && u === "\\" && (i = n - 1, o = !0, f += e.slice(a, i));
	}
	return f + e.slice(a, t - 1);
}
function F(e, n, t) {
	if (e === "true") return !0;
	if (e === "false") return !1;
	if (e === "-inf") return -Infinity;
	if (e === "inf" || e === "+inf") return Infinity;
	if (e === "nan" || e === "+nan" || e === "-nan") return NaN;
	if (e === "-0") return 0;
	let l;
	if ((l = M.test(e)) || Z.test(e)) {
		if (j.test(e)) throw new c("leading zeroes are not allowed", {
			toml: n,
			ptr: t
		});
		let i = +e.replace(/_/g, "");
		if (isNaN(i)) throw new c("invalid number", {
			toml: n,
			ptr: t
		});
		if (l && !Number.isSafeInteger(i)) throw new c("integer value cannot be represented losslessly", {
			toml: n,
			ptr: t
		});
		return i;
	}
	let r = new w(e);
	if (!r.isValid()) throw new c("invalid value", {
		toml: n,
		ptr: t
	});
	return r;
}
/*!
* Copyright (c) Squirrel Chat et al., All rights reserved.
* SPDX-License-Identifier: BSD-3-Clause
*
* Redistribution and use in source and binary forms, with or without
* modification, are permitted provided that the following conditions are met:
*
* 1. Redistributions of source code must retain the above copyright notice, this
*    list of conditions and the following disclaimer.
* 2. Redistributions in binary form must reproduce the above copyright notice,
*    this list of conditions and the following disclaimer in the
*    documentation and/or other materials provided with the distribution.
* 3. Neither the name of the copyright holder nor the names of its contributors
*    may be used to endorse or promote products derived from this software without
*    specific prior written permission.
*
* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
* ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
* DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
* FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
* DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
* CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
*/ function V(e, n, t, l) {
	let r = e.slice(n, t), i = r.indexOf("#");
	i > -1 && (y(e, i), r = r.slice(0, i));
	let o = r.trimEnd();
	if (!l) {
		let f = r.indexOf(`
`, o.length);
		if (f > -1) throw new c("newlines are not allowed in inline tables", {
			toml: e,
			ptr: n + f
		});
	}
	return [o, i];
}
function b(e, n, t, l) {
	if (l === 0) throw new c("document contains excessively nested structures. aborting.", {
		toml: e,
		ptr: n
	});
	let r = e[n];
	if (r === "[" || r === "{") {
		let [f, a] = r === "[" ? U(e, n, l) : K(e, n, l), u = A(e, a, ",", t);
		if (t === "}") {
			let d = g(e, a, u);
			if (d > -1) throw new c("newlines are not allowed in inline tables", {
				toml: e,
				ptr: d
			});
		}
		return [f, u];
	}
	let i;
	if (r === "\"" || r === "'") {
		i = S(e, n);
		let f = $(e, n, i);
		if (t) {
			if (i = s(e, i, t !== "]"), e[i] && e[i] !== "," && e[i] !== t && e[i] !== `
` && e[i] !== "\r") throw new c("unexpected character encountered", {
				toml: e,
				ptr: i
			});
			i += +(e[i] === ",");
		}
		return [f, i];
	}
	i = A(e, n, ",", t);
	let o = V(e, n, i - +(e[i - 1] === ","), t === "]");
	if (!o[0]) throw new c("incomplete key-value declaration: no value specified", {
		toml: e,
		ptr: n
	});
	return t && o[1] > -1 && (i = s(e, n + o[1]), i += +(e[i] === ",")), [F(o[0], e, n), i];
}
/*!
* Copyright (c) Squirrel Chat et al., All rights reserved.
* SPDX-License-Identifier: BSD-3-Clause
*
* Redistribution and use in source and binary forms, with or without
* modification, are permitted provided that the following conditions are met:
*
* 1. Redistributions of source code must retain the above copyright notice, this
*    list of conditions and the following disclaimer.
* 2. Redistributions in binary form must reproduce the above copyright notice,
*    this list of conditions and the following disclaimer in the
*    documentation and/or other materials provided with the distribution.
* 3. Neither the name of the copyright holder nor the names of its contributors
*    may be used to endorse or promote products derived from this software without
*    specific prior written permission.
*
* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
* ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
* DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
* FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
* DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
* CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
*/ let G = /^[a-zA-Z0-9-_]+[ \t]*$/;
function x(e, n, t = "=") {
	let l = n - 1, r = [], i = e.indexOf(t, n);
	if (i < 0) throw new c("incomplete key-value: cannot find end of key", {
		toml: e,
		ptr: n
	});
	do {
		let o = e[n = ++l];
		if (o !== " " && o !== "	") if (o === "\"" || o === "'") {
			if (o === e[n + 1] && o === e[n + 2]) throw new c("multiline strings are not allowed in keys", {
				toml: e,
				ptr: n
			});
			let f = S(e, n);
			if (f < 0) throw new c("unfinished string encountered", {
				toml: e,
				ptr: n
			});
			l = e.indexOf(".", f);
			let a = e.slice(f, l < 0 || l > i ? i : l), u = g(a);
			if (u > -1) throw new c("newlines are not allowed in keys", {
				toml: e,
				ptr: n + l + u
			});
			if (a.trimStart()) throw new c("found extra tokens after the string part", {
				toml: e,
				ptr: f
			});
			if (i < f && (i = e.indexOf(t, f), i < 0)) throw new c("incomplete key-value: cannot find end of key", {
				toml: e,
				ptr: n
			});
			r.push($(e, n, f));
		} else {
			l = e.indexOf(".", n);
			let f = e.slice(n, l < 0 || l > i ? i : l);
			if (!G.test(f)) throw new c("only letter, numbers, dashes and underscores are allowed in keys", {
				toml: e,
				ptr: n
			});
			r.push(f.trimEnd());
		}
	} while (l + 1 && l < i);
	return [r, s(e, i + 1, !0, !0)];
}
function K(e, n, t) {
	let l = {}, r = /* @__PURE__ */ new Set(), i, o = 0;
	for (n++; (i = e[n++]) !== "}" && i;) {
		if (i === `
`) throw new c("newlines are not allowed in inline tables", {
			toml: e,
			ptr: n - 1
		});
		if (i === "#") throw new c("inline tables cannot contain comments", {
			toml: e,
			ptr: n - 1
		});
		if (i === ",") throw new c("expected key-value, found comma", {
			toml: e,
			ptr: n - 1
		});
		if (i !== " " && i !== "	") {
			let f, a = l, u = !1, [d, N$2] = x(e, n - 1);
			for (let m = 0; m < d.length; m++) {
				if (m && (a = u ? a[f] : a[f] = {}), f = d[m], (u = Object.hasOwn(a, f)) && (typeof a[f] != "object" || r.has(a[f]))) throw new c("trying to redefine an already defined value", {
					toml: e,
					ptr: n
				});
				!u && f === "__proto__" && Object.defineProperty(a, f, {
					enumerable: !0,
					configurable: !0,
					writable: !0
				});
			}
			if (u) throw new c("trying to redefine an already defined value", {
				toml: e,
				ptr: n
			});
			let [_$2, k$1] = b(e, N$2, "}", t - 1);
			r.add(_$2), a[f] = _$2, n = k$1, o = e[n - 1] === "," ? n - 1 : 0;
		}
	}
	if (o) throw new c("trailing commas are not allowed in inline tables", {
		toml: e,
		ptr: o
	});
	if (!i) throw new c("unfinished table encountered", {
		toml: e,
		ptr: n
	});
	return [l, n];
}
function U(e, n, t) {
	let l = [], r;
	for (n++; (r = e[n++]) !== "]" && r;) {
		if (r === ",") throw new c("expected value, found comma", {
			toml: e,
			ptr: n - 1
		});
		if (r === "#") n = y(e, n);
		else if (r !== " " && r !== "	" && r !== `
` && r !== "\r") {
			let i = b(e, n - 1, "]", t - 1);
			l.push(i[0]), n = i[1];
		}
	}
	if (!r) throw new c("unfinished array encountered", {
		toml: e,
		ptr: n
	});
	return [l, n];
}
/*!
* Copyright (c) Squirrel Chat et al., All rights reserved.
* SPDX-License-Identifier: BSD-3-Clause
*
* Redistribution and use in source and binary forms, with or without
* modification, are permitted provided that the following conditions are met:
*
* 1. Redistributions of source code must retain the above copyright notice, this
*    list of conditions and the following disclaimer.
* 2. Redistributions in binary form must reproduce the above copyright notice,
*    this list of conditions and the following disclaimer in the
*    documentation and/or other materials provided with the distribution.
* 3. Neither the name of the copyright holder nor the names of its contributors
*    may be used to endorse or promote products derived from this software without
*    specific prior written permission.
*
* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
* ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
* DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
* FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
* DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
* CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
*/ function p(e, n, t, l) {
	let r = n, i = t, o, f = !1, a;
	for (let u = 0; u < e.length; u++) {
		if (u) {
			if (r = f ? r[o] : r[o] = {}, i = (a = i[o]).c, l === 0 && (a.t === 1 || a.t === 2)) return null;
			if (a.t === 2) {
				let d = r.length - 1;
				r = r[d], i = i[d].c;
			}
		}
		if (o = e[u], (f = Object.hasOwn(r, o)) && i[o]?.t === 0 && i[o]?.d) return null;
		f || (o === "__proto__" && (Object.defineProperty(r, o, {
			enumerable: !0,
			configurable: !0,
			writable: !0
		}), Object.defineProperty(i, o, {
			enumerable: !0,
			configurable: !0,
			writable: !0
		})), i[o] = {
			t: u < e.length - 1 && l === 2 ? 3 : l,
			d: !1,
			i: 0,
			c: {}
		});
	}
	if (a = i[o], a.t !== l && !(l === 1 && a.t === 3) || (l === 2 && (a.d || (a.d = !0, r[o] = []), r[o].push(r = {}), a.c[a.i++] = a = {
		t: 1,
		d: !1,
		i: 0,
		c: {}
	}), a.d)) return null;
	if (a.d = !0, l === 1) r = f ? r[o] : r[o] = {};
	else if (l === 0 && f) return null;
	return [
		o,
		r,
		a.c
	];
}
function X(e, n) {
	let t = n?.maxDepth ?? 1e3, l = {}, r = {}, i = l, o = r;
	for (let f = s(e, 0); f < e.length;) {
		if (e[f] === "[") {
			let a = e[++f] === "[", u = x(e, f += +a, "]");
			if (a) {
				if (e[u[1] - 1] !== "]") throw new c("expected end of table declaration", {
					toml: e,
					ptr: u[1] - 1
				});
				u[1]++;
			}
			let d = p(u[0], l, r, a ? 2 : 1);
			if (!d) throw new c("trying to redefine an already defined table or value", {
				toml: e,
				ptr: f
			});
			o = d[2], i = d[1], f = u[1];
		} else {
			let a = x(e, f), u = p(a[0], i, o, 0);
			if (!u) throw new c("trying to redefine an already defined table or value", {
				toml: e,
				ptr: f
			});
			let d = b(e, a[1], void 0, t);
			u[1][u[0]] = d[0], f = d[1];
		}
		if (f = s(e, f, !0), e[f] && e[f] !== `
` && e[f] !== "\r") throw new c("each key-value declaration must be followed by an end-of-line", {
			toml: e,
			ptr: f
		});
		f = s(e, f);
	}
	return l;
}
function Q(e) {
	const n = X(e);
	return N$1(e, n, { preserveIndentation: !1 }), n;
}

//#endregion
//#region node_modules/.pnpm/confbox@0.2.2/node_modules/confbox/dist/jsonc.mjs
var jsonc_exports = /* @__PURE__ */ __exportAll({ parseJSONC: () => h$1 });

//#endregion
export { yaml_exports as a, gr as i, Q as n, json5_exports as o, toml_exports as r, jsonc_exports as t };
